--- telepathy-qt-0.14.1/CMakeLists.txt.orig	2008-05-13 01:29:42.000000000 -0300
+++ telepathy-qt-0.14.1/CMakeLists.txt	2008-05-13 01:29:45.000000000 -0300
@@ -55,7 +55,7 @@ add_custom_target(uninstall
 
 add_subdirectory(src)
 add_subdirectory(include)
-add_subdirectory(tests)
+#add_subdirectory(tests)
 
 # If the cmake version includes cpack, use it
 IF(EXISTS "${CMAKE_ROOT}/Modules/CPack.cmake")
