Description: Fix FTBFS with GCC 6
Author: Alex Henrie
Origin: upstream, https://bazaar.launchpad.net/~inkscape.dev/inkscape/trunk/revision/15017
Bug: https://bugs.launchpad.net/inkscape/+bug/1524838
Bug-Debian: https://bugs.debian.org/835716
Acked-By: Mattia Rizzolo <mattia@debian.org>

--- a/src/ui/dialog/layer-properties.cpp
+++ b/src/ui/dialog/layer-properties.cpp
@@ -146,7 +146,7 @@
     destroy_();
     Glib::signal_idle().connect(
         sigc::bind_return(
-            sigc::bind(sigc::ptr_fun(&::operator delete), this),
+            sigc::bind(sigc::ptr_fun<void*, void>(&::operator delete), this),
             false 
         )
     );
