--- cups-1.3.5/configure.in	2007-07-24 23:47:12.000000000 +0000
+++ cups-1.3.5/configure.in.new	2008-05-23 22:18:31.000000000 +0000
@@ -42,6 +42,17 @@
 sinclude(config-scripts/cups-pdf.m4)
 sinclude(config-scripts/cups-scripting.m4)
 
+# Check for ucred
+AC_MSG_CHECKING(for struct ucred)
+AC_TRY_LINK([
+#include <sys/ucred.h>
+#include <sys/socket.h>
+],[
+struct ucred sucred; sucred.pid=0;
+],[has_ucred=yes],[has_ucred=no])
+AC_MSG_RESULT($has_ucred)
+if test ${has_ucred} = no ; then CFLAGS="$CFLAGS -DNEED_UCRED"; fi
+
 INSTALL_LANGUAGES=""
 UNINSTALL_LANGUAGES=""
 LANGFILES=""
--- cups-1.3.7/scheduler/auth.c	2008-05-23 22:25:23.000000000 +0000
+++ cups-1.3.7/scheduler/auth.c.new	2008-05-23 23:22:02.000000000 +0000
@@ -91,6 +91,14 @@
 typedef struct xucred cupsd_ucred_t;
 #  define CUPSD_UCRED_UID(c) (c).cr_uid
 #else
+#ifdef NEED_UCRED /* from autoconf */
+struct ucred
+   {
+      pid_t pid; /* PID of sending process. */
+      uid_t uid; /* UID of sending process. */
+      gid_t gid; /* GID of sending process. */
+   };
+#endif /* NEED_UCRED */
 typedef struct ucred cupsd_ucred_t;
 #  define CUPSD_UCRED_UID(c) (c).uid
 #endif /* HAVE_SYS_UCRED_H */
