--- lxc-2.0.8/configure.ac.orig	2017-05-31 23:10:49.773204490 -0300
+++ lxc-2.0.8/configure.ac	2017-05-31 23:11:37.517205386 -0300
@@ -345,14 +345,7 @@ fi
 
 # Check for static libcap, make sure the function checked for differs from the
 # the one checked below so the cache doesn't give a wrong answer
-OLD_CFLAGS="$CFLAGS"
-CFLAGS="$CFLAGS -static"
-AC_CHECK_LIB([cap],[cap_init],[have_static_libcap=yes],[have_static_libcap=no])
-AM_CONDITIONAL([HAVE_STATIC_LIBCAP], [test "x$have_static_libcap" = "xyes"])
-if test "x$have_static_libcap" = "xyes"; then
-	AC_DEFINE([HAVE_STATIC_LIBCAP], 1, [Have static libcap])
-fi
-CFLAGS="$OLD_CFLAGS"
+AM_CONDITIONAL([HAVE_STATIC_LIBCAP], [test "false" = "true"])
 
 
 # Linux capabilities
