diff -Nur pforth_v24/build/unix/Makefile pforth_v24.modified/build/unix/Makefile
--- pforth_v24/build/unix/Makefile	2009-02-20 19:38:25.000000000 +0100
+++ pforth_v24.modified/build/unix/Makefile	2010-05-13 19:40:06.801057855 +0200
@@ -101,3 +101,7 @@
 	-rm $(PFDICDAT)
 	-rm $(PFORTHDIC)
 	-rm $(PFDICAPP)
+
+INSTALL = /usr/bin/install -D
+install: all
+	$(INSTALL) ../../pforth_standalone $(DESTDIR)/bin/pforth_standalone
