diff -Naur gpart-0.1h/src/disku.c gpart-0.1h.new/src/disku.c
--- gpart-0.1h/src/disku.c	2001-02-07 20:04:07.000000000 +0100
+++ gpart-0.1h.new/src/disku.c	2008-02-07 15:32:24.000000000 +0100
@@ -16,6 +16,7 @@
  */
 
 
+#include <errno.h>
 #include <stdio.h>
 #include <string.h>
 #include <sys/ioctl.h>
@@ -27,7 +28,6 @@
 #endif
 
 #if defined(__FreeBSD__)
-#include <errno.h>
 #include <sys/disklabel.h>
 #endif
 
diff -Naur gpart-0.1h/src/l64seek.c gpart-0.1h.new/src/l64seek.c
--- gpart-0.1h/src/l64seek.c	2000-12-13 23:54:11.000000000 +0100
+++ gpart-0.1h.new/src/l64seek.c	2008-02-07 18:54:48.000000000 +0100
@@ -16,7 +16,6 @@
 
 #include "l64seek.h"
 
-
 #define OSTACKLEN	16
 static struct
 {
@@ -26,7 +25,12 @@
 static int		osptr = -1;
 
 #if defined(__linux__) && defined(__i386__)
-_syscall5(int,_llseek,uint,fd,ulong,hi,ulong,lo,loff_t *,res,uint,wh)
+static int _llseek (unsigned int fd, unsigned long oh,
+		    unsigned long ol, long long *result,
+		    unsigned int origin) {
+	return syscall (__NR__llseek, fd, oh, ol, result, origin);
+}
+//_syscall5(int,_llseek,uint,fd,ulong,hi,ulong,lo,loff_t *,res,uint,wh)
 #endif
 
 
