# fix a cmake pthread error
diff -Nur OpenSceneGraph/src/OpenThreads/pthreads/CMakeLists.txt OpenSceneGraph-fabio.m/src/OpenThreads/pthreads/CMakeLists.txt
--- OpenSceneGraph/src/OpenThreads/pthreads/CMakeLists.txt	2008-04-24 08:34:43.000000000 -0300
+++ OpenSceneGraph-fabio.m/src/OpenThreads/pthreads/CMakeLists.txt	2008-08-01 11:41:32.000000000 -0300
@@ -27,7 +27,7 @@
 ENDIF(OPENTHREADS_SONAMES)
 
 SET(CMAKE_REQUIRED_LIBRARIES_SAFE "${CMAKE_REQUIRED_LIBRARIES}")
-SET(CMAKE_REQUIRED_LIBRARIES "${CMAKE_REQUIRED_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT}")
+SET(CMAKE_REQUIRED_LIBRARIES ${CMAKE_REQUIRED_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
 
 CHECK_FUNCTION_EXISTS(pthread_yield HAVE_PTHREAD_YIELD)
 IF(HAVE_PTHREAD_YIELD)
