diff -Naur libdrm-2.4.17/include/drm/Makefile.am libdrm-2.4.17-new/include/drm/Makefile.am
--- libdrm-2.4.17/include/drm/Makefile.am	2009-12-01 00:17:08.000000000 +0100
+++ libdrm-2.4.17-new/include/drm/Makefile.am	2010-01-05 16:38:42.000000000 +0100
@@ -22,7 +22,7 @@
 # however, r300 and via need their reg headers installed in order to build.
 # better solutions are welcome.
 
-klibdrmincludedir = ${includedir}/drm
+klibdrmincludedir = ${includedir}/xorg/drm
 klibdrminclude_HEADERS = \
 	drm.h \
 	drm_mode.h \
diff -Naur libdrm-2.4.17/include/drm/Makefile.in libdrm-2.4.17-new/include/drm/Makefile.in
--- libdrm-2.4.17/include/drm/Makefile.in	2009-12-21 06:02:29.000000000 +0100
+++ libdrm-2.4.17-new/include/drm/Makefile.in	2010-01-05 16:38:34.000000000 +0100
@@ -218,7 +218,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-klibdrmincludedir = ${includedir}/drm
+klibdrmincludedir = ${includedir}/xorg/drm
 klibdrminclude_HEADERS = \
 	drm.h \
 	drm_mode.h \
diff -Naur libdrm-2.4.17/intel/libdrm_intel.pc.in libdrm-2.4.17-new/intel/libdrm_intel.pc.in
--- libdrm-2.4.17/intel/libdrm_intel.pc.in	2009-12-01 00:17:08.000000000 +0100
+++ libdrm-2.4.17-new/intel/libdrm_intel.pc.in	2010-01-05 16:39:43.000000000 +0100
@@ -7,4 +7,4 @@
 Description: Userspace interface to kernel DRM services
 Version: @PACKAGE_VERSION@
 Libs: -L${libdir} -ldrm -ldrm_intel
-Cflags: -I${includedir} -I${includedir}/drm
+Cflags: -I${includedir} -I${includedir}/xorg/drm
diff -Naur libdrm-2.4.17/libdrm.pc.in libdrm-2.4.17-new/libdrm.pc.in
--- libdrm-2.4.17/libdrm.pc.in	2007-08-16 13:30:30.000000000 +0200
+++ libdrm-2.4.17-new/libdrm.pc.in	2010-01-05 16:39:04.000000000 +0100
@@ -7,4 +7,4 @@
 Description: Userspace interface to kernel DRM services
 Version: @PACKAGE_VERSION@
 Libs: -L${libdir} -ldrm
-Cflags: -I${includedir} -I${includedir}/drm
+Cflags: -I${includedir} -I${includedir}/xorg/drm
