--- faad2/bootstrap.orig	2007-02-14 13:23:39.000000000 +0000
+++ faad2/bootstrap	2007-02-14 13:24:16.000000000 +0000
@@ -1,17 +1,17 @@
-#! /bin/sh
-
-case $OSTYPE in
-	darwin*)
-	LIBTOOLIZE=glibtoolize
-	;;
-	*)
-	LIBTOOLIZE=libtoolize
-	;;
-esac
-
-aclocal -I .                   && \
-autoheader                     && \
-$LIBTOOLIZE --automake --copy   && \
-automake --add-missing --copy  && \
-autoconf                       && \
-echo "Ready to run ./configure"
+#! /bin/sh
+
+case $OSTYPE in
+	darwin*)
+	LIBTOOLIZE=glibtoolize
+	;;
+	*)
+	LIBTOOLIZE=libtoolize
+	;;
+esac
+
+aclocal -I .
+autoheader
+$LIBTOOLIZE --automake --copy
+automake --add-missing --copy
+autoconf
+echo "Ready to run ./configure"
