--- boost_1_34_1/tools/jam/src/build.jam.orig	2008-02-07 09:09:04.000000000 +0100
+++ boost_1_34_1/tools/jam/src/build.jam	2008-02-07 09:09:41.000000000 +0100
@@ -152,7 +152,7 @@
     : -L$(--python-lib[1]) -l$(--python-lib[2]) ;
 ## GCC 2.x, 3.x, 4.x
 toolset gcc gcc : "-o " : -D
-    : -pedantic
+    : -pedantic -fno-strict-aliasing
     [ opt --release : [ opt --symbols : -g : -s ] -O3 ]
     [ opt --debug : -g -O0 -fno-inline ]
     -I$(--python-include) -Wno-long-long
