--- kdebase-3.5.6/konqueror/keditbookmarks/Makefile.am.orig	2007-03-26 02:34:38.000000000 -0300
+++ kdebase-3.5.6/konqueror/keditbookmarks/Makefile.am	2007-03-26 02:34:46.000000000 -0300
@@ -13,7 +13,7 @@ kbookmarkmerger_LDADD = $(LIB_KIO)
 dcop_DCOPIDLNG = true
 keditbookmarks_la_SOURCES = main.cpp listview.cpp toplevel.cpp actionsimpl.cpp commands.cpp importers.cpp dcop.skel dcop.cpp bookmarkiterator.cpp  \
                             testlink.cpp favicons.cpp updater.cpp exporters.cpp kbookmarknotifier.stub bookmarkinfo.cpp kebsearchline.cpp settings.kcfgc
-kbookmarknotifier_DIR = $(includedir)
+kbookmarknotifier_DIR = $(kde_includes)
 keditbookmarks_la_LIBADD  = $(top_builddir)/libkonq/libkonq.la
 keditbookmarks_la_LDFLAGS = $(all_libraries) -module -avoid-version
 
--- kdebase-3.5.6/konqueror/keditbookmarks/Makefile.in.orig	2007-03-26 02:34:42.000000000 -0300
+++ kdebase-3.5.6/konqueror/keditbookmarks/Makefile.in	2007-03-26 02:34:46.000000000 -0300
@@ -622,7 +622,7 @@ keditbookmarks_la_SOURCES = keditbookmar
 libkdeinit_keditbookmarks_la_SOURCES = main.cpp listview.cpp toplevel.cpp actionsimpl.cpp commands.cpp importers.cpp  dcop.cpp bookmarkiterator.cpp  \
                             testlink.cpp favicons.cpp updater.cpp exporters.cpp  bookmarkinfo.cpp kebsearchline.cpp  dcop_skel.cpp kbookmarknotifier_stub.cpp settings.cpp
 
-kbookmarknotifier_DIR = $(includedir)
+kbookmarknotifier_DIR = $(kde_includes)
 #>- keditbookmarks_la_LIBADD = $(top_builddir)/libkonq/libkonq.la
 #>+ 1
 keditbookmarks_la_LIBADD = libkdeinit_keditbookmarks.la
@@ -1097,8 +1097,8 @@ dcop.kidl: $(srcdir)/dcop.h $(DCOP_DEPEN
 	KDECONFIG="$(KDECONFIG)" $(DCOPIDLNG) $(srcdir)/dcop.h > dcop.kidl || ( rm -f dcop.kidl ; false )
 dcop_skel.cpp: dcop.kidl
 	$(DCOPIDL2CPP) --c++-suffix cpp --no-signals --no-stub dcop.kidl
-kbookmarknotifier.kidl: $(includedir)/kbookmarknotifier.h $(DCOP_DEPENDENCIES)
-	$(DCOPIDL) $(includedir)/kbookmarknotifier.h > kbookmarknotifier.kidl || ( rm -f kbookmarknotifier.kidl ; false )
+kbookmarknotifier.kidl: $(kde_includes)/kbookmarknotifier.h $(DCOP_DEPENDENCIES)
+	$(DCOPIDL) $(kde_includes)/kbookmarknotifier.h > kbookmarknotifier.kidl || ( rm -f kbookmarknotifier.kidl ; false )
 kbookmarknotifier_stub.h: kbookmarknotifier.kidl
 	$(DCOPIDL2CPP) --c++-suffix cpp --no-signals --no-skel kbookmarknotifier.kidl
 
