diff -Naur ndiswrapper-1.2.old/driver/Makefile ndiswrapper-1.2/driver/Makefile
--- ndiswrapper-1.2.old/driver/Makefile	2005-06-10 21:24:31.000000000 +0200
+++ ndiswrapper-1.2/driver/Makefile	2005-07-27 11:38:21.000000000 +0200
@@ -12,7 +12,7 @@
 KPSUB := $(shell echo $(KVERS) |  sed -e 's/\([^\.]*\)\.\([^\.]*\)\..*/\1\2/')
 
 DESTDIR =
-INST_DIR := $(DESTDIR)/lib/modules/$(KVERS)/misc
+INST_DIR := $(goboModules)/$(KVERS)/misc
 
 SRC_DIR=$(shell pwd)
 
diff -Naur ndiswrapper-1.2.old/Makefile ndiswrapper-1.2/Makefile
--- ndiswrapper-1.2.old/Makefile	2005-06-10 21:24:31.000000000 +0200
+++ ndiswrapper-1.2/Makefile	2005-07-27 11:29:37.000000000 +0200
@@ -7,7 +7,7 @@
 DIST_SUBDIRS=utils driver debian
 
 DESTDIR = 
-mandir = $(DESTDIR)$(shell [ -d /usr/man/man8 ] && echo /usr/man || echo /usr/share/man )
+mandir = $(DESTDIR)$(shell [ -d /man/man8 ] && echo /man || echo /share/man )
 
 KVERS ?= $(shell uname -r)
 
diff -Naur ndiswrapper-1.2.old/utils/Makefile ndiswrapper-1.2/utils/Makefile
--- ndiswrapper-1.2.old/utils/Makefile	2005-06-10 21:24:31.000000000 +0200
+++ ndiswrapper-1.2/utils/Makefile	2005-07-27 11:39:26.000000000 +0200
@@ -2,7 +2,7 @@
 
 DESTDIR =
 sbindir = $(DESTDIR)/sbin
-usrsbindir = $(DESTDIR)/usr/sbin
+usrsbindir = $(DESTDIR)/sbin
 
 OBJS = loadndisdriver.o
 
