Unregisters applesmc_led on exit.

Signed-off-by: Lucas C. Villa Real <lucasvr@gobolinux.org>

--- linux-2.6.24/drivers/hwmon/applesmc.c.orig	2008-02-04 06:32:24.000000000 -0200
+++ linux-2.6.24/drivers/hwmon/applesmc.c	2008-02-04 06:32:26.000000000 -0200
@@ -1409,6 +1409,7 @@ static void __exit applesmc_exit(void)
 		destroy_workqueue(applesmc_led_wq);
 		sysfs_remove_file(&pdev->dev.kobj, &dev_attr_light.attr);
 	}
+	led_classdev_unregister(&applesmc_led);
 	if (applesmc_accelerometer)
 		applesmc_release_accelerometer();
 	sysfs_remove_group(&pdev->dev.kobj, &temperature_attributes_group);
