--- openbabel-2.1.1/include/openbabel/xml.h.orig	2008-02-25 23:44:36.000000000 
+0100
+++ openbabel-2.1.1/include/openbabel/xml.h	2008-02-25 23:44:20.000000000 +0100
@@ -19,6 +19,7 @@ GNU General Public License for more deta
 
 #include <libxml/xmlreader.h>
 #include <libxml/xmlwriter.h>
+#include <typeinfo>
 
 namespace OpenBabel
 {
--- openbabel-2.1.1/include/openbabel/obmolecformat.h.orig	2008-02-25 
23:44:55.000000000 +0100
+++ openbabel-2.1.1/include/openbabel/obmolecformat.h	2008-02-25 23:45:07.000000000 
+0100
@@ -25,6 +25,7 @@ GNU General Public License for more deta
 
 #include <openbabel/mol.h>
 #include <openbabel/obconversion.h>
+#include <typeinfo>
 
 namespace OpenBabel {
 
--- openbabel-2.1.1/include/openbabel/data.h.orig	2008-02-25 23:45:32.000000000 
+0100
+++ openbabel-2.1.1/include/openbabel/data.h	2008-02-25 23:45:52.000000000 +0100
@@ -27,6 +27,7 @@ GNU General Public License for more deta
 #include <fstream>
 #include <vector>
 #include <string>
+#include <cstring>
 
 namespace OpenBabel
 {
--- openbabel-2.1.1/include/openbabel/lineend.h.orig	2008-02-25 23:46:03.000000000 
+0100
+++ openbabel-2.1.1/include/openbabel/lineend.h	2008-02-25 23:46:15.000000000 
+0100
@@ -21,6 +21,7 @@ General Public License for more details.
 #define OB_LINEEND_H
 
 #include <streambuf>
+#include <climits>
 
 #ifndef OBCONV
 #define OBCONV
--- openbabel-2.1.1/src/residue.cpp.orig	2008-02-25 23:40:15.000000000 +0100
+++ openbabel-2.1.1/src/residue.cpp	2008-02-25 23:44:02.000000000 +0100
@@ -33,6 +33,8 @@ obtained in part or whole from RasMol2 b
 #include <openbabel/oberror.h>
 #include <openbabel/bitvec.h>
 #include <openbabel/bond.h>
+#include <cstring>
+#include <cstdlib>
 
 using namespace std;
 
--- openbabel-2.1.1/src/tokenst.cpp.orig	2008-02-25 23:40:44.000000000 +0100
+++ openbabel-2.1.1/src/tokenst.cpp	2008-02-25 23:41:02.000000000 +0100
@@ -25,6 +25,7 @@ GNU General Public License for more deta
 #include <algorithm>
 #include <vector>
 #include <string>
+#include <cstring>
 
 
 using namespace std;
--- openbabel-2.1.1/src/dlhandler_unix.cpp.orig	2008-02-25 23:41:16.000000000 
+0100
+++ openbabel-2.1.1/src/dlhandler_unix.cpp	2008-02-25 23:41:35.000000000 +0100
@@ -25,6 +25,8 @@ GNU General Public License for more deta
 #include <sys/stat.h>
 #include <dlfcn.h>
 #include <cstdlib>
+#include <cstring>
+
 
 using namespace std;
 
--- openbabel-2.1.1/src/fingerprint.cpp.orig	2008-02-26 00:01:56.000000000 +0100
+++ openbabel-2.1.1/src/fingerprint.cpp	2008-02-26 00:02:16.000000000 +0100
@@ -23,6 +23,7 @@ GNU General Public License for more deta
 #include <iosfwd>
 #include <vector>
 #include <fstream>
+#include <cstring>
 
 #include <openbabel/fingerprint.h>
 #include <openbabel/oberror.h>
--- openbabel-2.1.1/include/openbabel/obconversion.h.orig	2008-02-26 
00:09:59.000000000 +0100
+++ openbabel-2.1.1/include/openbabel/obconversion.h	2008-02-26 00:10:20.000000000 
+0100
@@ -29,6 +29,8 @@ GNU General Public License for more deta
 #include <vector>
 #include <map>
 
+#include <cstring>
+
 #include <openbabel/dlhandler.h>
 #include <openbabel/oberror.h>
 #include <openbabel/lineend.h>
--- openbabel-2.1.1/src/zipstreamimpl.h.orig	2008-02-26 00:10:43.000000000 +0100
+++ openbabel-2.1.1/src/zipstreamimpl.h	2008-02-26 00:11:02.000000000 +0100
@@ -31,6 +31,8 @@ Altered by: Geoffrey Hutchison 2005 for 
             minor namespace modifications, VC++ compatibility
 */
 
+#include <cstring>
+
 
 //*****************************************************************************
 //  template class basic_zip_streambuf
 //*****************************************************************************
--- openbabel-2.1.1/src/obconversion.cpp.orig	2008-02-26 00:12:12.000000000 +0100
+++ openbabel-2.1.1/src/obconversion.cpp	2008-02-26 00:12:25.000000000 +0100
@@ -37,6 +37,7 @@ GNU General Public License for more deta
 #include <sstream>
 #include <string>
 #include <map>
+#include <typeinfo>
 
 #include <openbabel/obconversion.h>
 
--- openbabel-2.1.1/src/formats/rxnformat.cpp.orig	2008-02-26 00:39:29.000000000 
+0100
+++ openbabel-2.1.1/src/formats/rxnformat.cpp	2008-02-26 00:39:47.000000000 +0100
@@ -18,6 +18,8 @@ GNU General Public License for more deta
 #pragma warning (disable : 4251) //
 #endif
 
+#include <typeinfo>
+
 #include <openbabel/babelconfig.h>
 
 #include <openbabel/mol.h>

