--- e2fsprogs-1.40.5/misc/Makefile.orig	2008-01-28 09:00:24.000000000 +0100
+++ e2fsprogs-1.40.5/misc/Makefile.in	2008-01-28 09:00:51.000000000 +0100
@@ -251,8 +251,7 @@
 		$(DESTDIR)$(root_sbindir) $(DESTDIR)$(bindir) \
 		$(DESTDIR)$(man1dir) $(DESTDIR)$(man8dir) \
 		$(DESTDIR)$(man1dir) $(DESTDIR)$(man5dir) \
-		$(DESTDIR)$(libdir) $(DESTDIR)/$(root_sysconfdir) \
-		$(DESTDIR)/etc/init.d
+		$(DESTDIR)$(libdir) $(DESTDIR)/$(root_sysconfdir) 
 
 install: all $(SMANPAGES) $(UMANPAGES) installdirs
 	@for i in $(SPROGS); do \
