--- syscall.c.orig	2008-08-24 23:16:26.000000000 -0400
+++ syscall.c	2009-03-08 15:31:34.000000000 -0400
@@ -1016,6 +1016,15 @@
 		/*
 		 * Note: we only deal with only 32-bit CPUs here.
 		 */
+
+		if (!(tcp->flags & TCB_INSYSCALL) &&
+		    (tcp->flags & TCB_WAITEXECVE)) {
+			/* caught a fake syscall from the execve's exit */
+			tcp->flags &= ~TCB_WAITEXECVE;
+			return 0;
+		}
+ 
+
 		if (regs.ARM_cpsr & 0x20) {
 			/*
 			 * Get the Thumb-mode system call number
