Don't call depmod either backup original modules.

--- kvm-64/kernel/Makefile	2008-03-26 13:49:35.000000000 +0100
+++ kvm-64.new/kernel/Makefile	2008-04-04 16:33:43.000000000 +0200
@@ -61,10 +61,6 @@
 install:
 	mkdir -p $(DESTDIR)/$(INSTALLDIR)
 	cp *.ko $(DESTDIR)/$(INSTALLDIR)
-	for i in $(ORIGMODDIR)/drivers/kvm/*.ko; do \
-		if [ -f "$$i" ]; then mv "$$i" "$$i.orig"; fi; \
-	done
-	/sbin/depmod -a
 
 tmpspec = .tmp.kvm-kmod.spec
 
