--- lm_sensors-3.0.0/Makefile.orig	2008-01-12 22:27:54.000000000 +0100
+++ lm_sensors-3.0.0/Makefile	2008-01-12 22:28:11.000000000 +0100
@@ -38,14 +38,14 @@
 DESTDIR :=
 
 # This is the prefix that will be used for almost all directories below.
-PREFIX := /usr/local
+PREFIX ?= /usr/local
 
 # Your C compiler
 CC := gcc
 
 # This is the directory where sensors.conf will be installed, if no other
 # configuration file is found
-ETCDIR := /etc
+ETCDIR ?= /etc
 
 # You should not need to change this. It is the directory into which the
 # library files (both static and shared) will be installed.
