--- strace-4.5.8/resource.c.orig	2006-02-17 17:28:41.000000000 -0200
+++ strace-4.5.8/resource.c	2006-02-17 17:30:06.000000000 -0200
@@ -475,11 +475,19 @@ static const struct xlat quotacmds[] = {
 	{ Q_QUOTAOFF,	"Q_QUOTAOFF"	},
 	{ Q_GETQUOTA,	"Q_GETQUOTA"	},
 	{ Q_SETQUOTA,	"Q_SETQUOTA"	},
+#ifdef Q_SETUSE
 	{ Q_SETUSE,	"Q_SETUSE"	},
+#endif
 	{ Q_SYNC,	"Q_SYNC"	},
+#ifdef Q_SETQLIM
 	{ Q_SETQLIM,	"Q_SETQLIM"	},
+#endif
+#ifdef Q_GETSTATS
 	{ Q_GETSTATS,	"Q_GETSTATS"	},
+#endif
+#ifdef Q_RSQUASH
 	{ Q_RSQUASH,	"Q_RSQUASH"	},
+#endif
 	{ NEW_CMD(0x1), "Q_SYNC"        },
 	{ NEW_CMD(0x2), "Q_QUOTAON"     },
 	{ NEW_CMD(0x3), "Q_QUOTAOFF"    },
