--- rox/install.sh	2003-11-25 23:06:04.000000000 -0200
+++ rox/patched_install.sh	2004-01-09 23:21:57.000000000 -0200
@@ -49,6 +49,15 @@
 
 umask 022
 
+if [ "$1" == "--prefix" ]; then
+    [ "$2" ] || die "Missing argument after --prefix"
+    DIR=$2
+    APPDIR="$DIR/apps"
+    BINDIR="$DIR/bin"
+    SHAREDIR="$DIR/share"
+    CHOICESDIR=${SHAREDIR}/Choices
+    MANDIR="$DIR/man"
+else
 
 cat << EOF
 
@@ -133,6 +142,8 @@
 	*) die "Invalid choice!";;
 esac
 
+fi
+
 MIMEDIR=${SHAREDIR}/mime
 
 MIMEINFO="${MIMEDIR}/packages/rox.xml"
