--- pinentry/gtk+-2/pinentry-gtk-2.c
+++ pinentry/gtk+-2/pinentry-gtk-2.c
@@ -261,10 +261,10 @@ create_window (int confirm_mode)
   if (!confirm_mode)
     {
       g_signal_connect (G_OBJECT (win),
-			pinentry->grab ? "map-event" : "focus-in-event",
+			"focus-in-event",
 			G_CALLBACK (grab_keyboard), NULL);
       g_signal_connect (G_OBJECT (win),
-			pinentry->grab ? "unmap-event" : "focus-out-event",
+			"focus-out-event",
 			G_CALLBACK (ungrab_keyboard), NULL);
     }
   gtk_window_add_accel_group (GTK_WINDOW (win), acc);
