--- gnupg-2.0.9/keyserver/gpgkeys_curl.c.old	2008-07-13 11:45:30.000000000 +0000
+++ gnupg-2.0.9/keyserver/gpgkeys_curl.c	2008-07-13 11:45:12.000000000 +0000
@@ -300,7 +300,7 @@
       curl_easy_setopt(curl,CURLOPT_VERBOSE,1);
     }
 
-  curl_easy_setopt(curl,CURLOPT_SSL_VERIFYPEER,opt->flags.check_cert);
+  curl_easy_setopt(curl,CURLOPT_SSL_VERIFYPEER,opt->flags.check_cert != 0);
   curl_easy_setopt(curl,CURLOPT_CAINFO,opt->ca_cert_file);
 
   if(proxy)
