--- gparted-0.3.5/src/GParted_Core.cc	2008-02-04 14:35:47.000000000 +0000
+++ gparted-0.3.5/src/GParted_Core.cc.new	2008-05-25 20:53:41.000000000 +0000
@@ -41,6 +41,7 @@
 #include <cstdio>
 #include <csignal>
 #include <sys/statvfs.h>	
+#include <cstdlib>
 
 std::vector<Glib::ustring> libparted_messages ; //see ped_exception_handler()
 
@@ -1711,7 +1712,7 @@
 
 	while ( succes &&
 		timer .elapsed() <= smallest_time && 
-		std::llabs( done ) + N <= length && 
+		llabs( done ) + N <= length && 
 		optimal_blocksize * 2 < N )
 	{
 		if ( done != 0 ) 
@@ -1755,7 +1756,7 @@
 		      	    	    dst_device,
 		      	    	    src_start + ( dst_start > src_start ? 0 : done ),
 		      	    	    dst_start + ( dst_start > src_start ? 0 : done ),
-		      	    	    length - std::llabs( done ), 
+		      	    	    length - llabs( done ), 
 		      	    	    optimal_blocksize,
 		      	    	    operationdetail,
 			    	    readonly,
@@ -1929,7 +1930,7 @@
 	if ( lp_device_src && lp_device_dst && ped_device_open( lp_device_src ) && ped_device_open( lp_device_dst ) )
 	{
 		Glib::ustring error_message ;
-		buf = static_cast<char *>( malloc( std::llabs( blocksize ) * 512 ) ) ;
+		buf = static_cast<char *>( malloc( llabs( blocksize ) * 512 ) ) ;
 		if ( buf )
 		{
 			ped_device_sync( lp_device_dst ) ;
@@ -1950,7 +1951,7 @@
 			operationdetail .get_last_child() .add_child( OperationDetail( "", STATUS_NONE ) ) ;
 			
 			Glib::Timer timer_progress_timeout, timer_total ;
-			while( succes && std::llabs( done ) < length )
+			while( succes && llabs( done ) < length )
 			{
 				succes = copy_block( lp_device_src,
 						     lp_device_dst,
@@ -1965,7 +1966,7 @@
 				if ( timer_progress_timeout .elapsed() >= 0.5 )
 				{
 					set_progress_info( length,
-							   std::llabs( done + blocksize ),
+							   llabs( done + blocksize ),
 							   timer_total,
 							   operationdetail .get_last_child() .get_last_child(),
 							   readonly ) ;
@@ -1984,13 +1985,13 @@
 
 		//final description
 		operationdetail .get_last_child() .get_last_child() .set_description( 
-			String::ucompose( readonly ? _("%1 of %2 read") : _("%1 of %2 copied"), std::llabs( done ), length ), FONT_ITALIC ) ;
+			String::ucompose( readonly ? _("%1 of %2 read") : _("%1 of %2 copied"), llabs( done ), length ), FONT_ITALIC ) ;
 		
 		if ( ! succes && ! error_message .empty() )
 			operationdetail .get_last_child() .add_child( 
 				OperationDetail( error_message, STATUS_NONE, FONT_ITALIC ) ) ;
 		
-		total_done += std::llabs( done ) ;
+		total_done += llabs( done ) ;
 	
 		//close and destroy the devices..
 		ped_device_close( lp_device_src ) ;
@@ -2017,7 +2018,7 @@
 {
 	if ( blocksize < 0 )
 	{
-		blocksize = std::llabs( blocksize ) ;
+		blocksize = llabs( blocksize ) ;
 		offset_src -= ( blocksize -1 ) ;
 		offset_dst -= ( blocksize -1 ) ;
 	}
