diff -Naur libmusicbrainz-3.0.1-orig/src/utils_private.cpp libmusicbrainz-3.0.1/src/utils_private.cpp
--- libmusicbrainz-3.0.1-orig/src/utils_private.cpp	2006-09-05 02:59:44.000000000 -0600
+++ libmusicbrainz-3.0.1/src/utils_private.cpp	2008-01-02 21:15:25.000000000 -0600
@@ -20,6 +20,7 @@
  * $Id: utils_private.cpp 8466 2006-09-05 08:59:44Z luks $
  */
  
+#include <cstdlib>
 #include <cstring>
 #include <cstdarg>
 #include <iostream>
diff -Naur libmusicbrainz-3.0.1-orig/src/webservice.cpp libmusicbrainz-3.0.1/src/webservice.cpp
--- libmusicbrainz-3.0.1-orig/src/webservice.cpp	2007-05-11 16:55:10.000000000 -0600
+++ libmusicbrainz-3.0.1/src/webservice.cpp	2008-01-02 21:12:27.000000000 -0600
@@ -24,6 +24,7 @@
 #include <string>
 #include <map>
 #include <iostream>
+#include <stdlib.h>
 #include <string.h>
 #include <ne_session.h>
 #include <ne_request.h>
