--- sqlite-3.3.17/Makefile.in.orig	2007-06-18 18:24:55.000000000 -0300
+++ sqlite-3.3.17/Makefile.in	2007-06-18 18:25:05.000000000 -0300
@@ -395,7 +395,7 @@ parse.lo:	parse.c $(HDR)
 
 parse.h:	parse.c
 
-parse.c:	$(TOP)/src/parse.y lemon$(BEXE) $(TOP)/addopcodes.awk
+parse.c:	$(TOP)/src/parse.y $(TOP)/addopcodes.awk
 	cp $(TOP)/src/parse.y .
 	./lemon$(BEXE) $(OPTS) parse.y
 	mv parse.h parse.h.temp
