--- udev-053-orig/etc/udev/udev.conf.in	2005-02-19 22:56:10.000000000 +0100
+++ udev-053/etc/udev/udev.conf.in	2005-02-19 23:00:15.000000000 +0100
@@ -17,3 +17,15 @@
 # udev_log - set to "yes" if you want logging, else "no"
 udev_log="yes"
 
+# The default permissions for all nodes that have no explicit match in the
+# permissions configuration files.
+default_mode="0660"
+
+# default_owner - set the default owner for all nodes that have no
+# explicit match in the permissions file
+default_owner="0"
+
+# default_group - set the default group for all nodes that have no
+# explicit match in the permissions file
+default_group="0"
+
--- udev-053-orig/etc/udev/udev.rules	2005-02-19 22:56:10.000000000 +0100
+++ udev-053/etc/udev/udev.rules	2005-02-20 00:21:14.000000000 +0100
@@ -5,43 +5,138 @@
 #
 
 # if this is a ide cdrom, name it the default name, and create a symlink to cdrom
-BUS="ide", KERNEL="*[!0-9]", PROGRAM="/bin/cat /proc/ide/%k/media", RESULT="cdrom", NAME="%k", SYMLINK="cdrom"
+BUS="ide", KERNEL="*[!0-9]", PROGRAM="/bin/cat /proc/ide/%k/media", RESULT="cdrom", NAME="%k", SYMLINK="cdrom", GROUP="cdrom"
 
 # create a symlink named after the device map name
 # note devmap_name comes with extras/multipath
-KERNEL="dm-[0-9]*", PROGRAM="/sbin/devmap_name %M %m", NAME="%k", SYMLINK="%c"
+KERNEL="dm-[0-9]*", PROGRAM="/sbin/devmap_name %M %m", NAME="%k", SYMLINK="%c", GROUP="disk", MODE="0640"
 
 # DRI devices always go into a subdirectory (as per the LSB spec)
-KERNEL="card*", NAME="dri/card%n"
+KERNEL="card*", NAME="dri/card%n", GROUP="video"
 
 # alsa devices
-KERNEL="controlC[0-9]*", NAME="snd/%k"
-KERNEL="hw[CD0-9]*",     NAME="snd/%k"
-KERNEL="pcm[CD0-9cp]*",  NAME="snd/%k"
-KERNEL="midiC[D0-9]*",   NAME="snd/%k"
-KERNEL="timer",          NAME="snd/%k"
-KERNEL="seq",            NAME="snd/%k"
+KERNEL="controlC[0-9]*", NAME="snd/%k", GROUP="audio"
+KERNEL="hw[CD0-9]*",     NAME="snd/%k", GROUP="audio"
+KERNEL="pcm[CD0-9cp]*",  NAME="snd/%k", GROUP="audio"
+KERNEL="midiC[D0-9]*",   NAME="snd/%k", GROUP="audio"
+KERNEL="timer",          NAME="snd/%k", GROUP="audio"
+KERNEL="seq",            NAME="snd/%k", GROUP="audio"
 
 # input devices
-KERNEL="mice",		NAME="input/%k"
-KERNEL="mouse*",	NAME="input/%k"
-KERNEL="event*",	NAME="input/%k"
-KERNEL="js*",		NAME="input/%k"
-KERNEL="ts*",		NAME="input/%k"
+KERNEL="mice",		NAME="input/%k", MODE="0644"
+KERNEL="mouse*",	NAME="input/%k", MODE="0644"
+KERNEL="event*",	NAME="input/%k", MODE="0644"
+KERNEL="js*",		NAME="input/%k", MODE="0644"
+KERNEL="ts*",		NAME="input/%k", MODE="0644"
 
 # USB devices
 KERNEL="hiddev*",	NAME="usb/%k"
 KERNEL="auer*",		NAME="usb/%k"
 KERNEL="legousbtower*",	NAME="usb/%k"
 KERNEL="dabusb*",	NAME="usb/%k"
-BUS="usb", KERNEL="lp[0-9]*",	NAME="usb/%k"
+BUS="usb", KERNEL="lp[0-9]*",	NAME="usb/%k", GROUP="lp"
 
 # CAPI devices
-KERNEL="capi",		NAME="capi20", SYMLINK="isdn/capi20"
-KERNEL="capi*",		NAME="capi/%n"
+KERNEL="capi",		NAME="capi20", SYMLINK="isdn/capi20", GROUP="dialout"
+KERNEL="capi*",		NAME="capi/%n", GROUP="dialout"
 
 # Network devices
 KERNEL="tun",		NAME="net/%k"
 
 # raw devices
-KERNEL="raw[0-9]*",	NAME="raw/%k" 
+KERNEL="raw[0-9]*",	NAME="raw/%k", GROUP="disk"
+
+KERNEL="ptmx", NAME="%k", GROUP="tty", MODE="0666"
+KERNEL="random", NAME="%k", MODE="0666"
+KERNEL="urandom", NAME="%k", MODE="0444"
+KERNEL="kmem", NAME="%k", GROUP="kmem", MODE="0640"
+KERNEL="mem", NAME="%k", GROUP="kmem", MODE="0640"
+KERNEL="port", NAME="%k", GROUP="kmem", MODE="0640"
+KERNEL="null", NAME="%k", MODE="0666"
+KERNEL="zero", NAME="%k", MODE="0666"
+KERNEL="full", NAME="%k", MODE="0666"
+KERNEL="aio", NAME=%k", MODE="0444"
+KERNEL="kmsg", NAME="%k", MODE="0600"
+
+KERNEL="ttyS[0-9]*", NAME="%k", GROUP="dialout"
+KERNEL="ttyUSB[0-9]*", NAME="%k", GROUP="dialout"
+KERNEL="rfcomm[0-9]*", NAME="%k", GROUP="dialout"
+KERNEL="tty[BCDEFHILMPRSTUVWX][0-9]*", NAME="%k", GROUP="dialout"
+KERNEL="ttyS[ACIR][0-9]*", NAME="%k", GROUP="dialout"
+KERNEL="ttyUSB[0-9]*", NAME="%k", GROUP="dialout"
+KERNEL="ttyACM[0-9]*", NAME="%k", GROUP="dialout"
+KERNEL="ippp[0-9]*", NAME="%k", GROUP="dialout"
+KERNEL="isdn[0-9]*", NAME="%k", GROUP="dialout"
+KERNEL="isdnctrl[0-9]*", NAME="%k", GROUP="dialout"
+KERNEL="dcbri[0-9]*", NAME="%k", GROUP="dialout"
+KERNEL="ircomm[0-9]*", NAME="%k", GROUP="dialout"
+
+KERNEL="tty", NAME="%k", GROUP="tty", MODE="0666"
+KERNEL="tty[0-9]*", NAME="%k", GROUP="tty", MODE=0666"
+KERNEL="vcs*", NAME="%k", MODE="0600"
+KERNEL="console", NAME="%k", GROUP="tty", MODE="0622"
+
+KERNEL="admmidi*", NAME="%k", GROUP="audio"
+KERNEL="adsp*", NAME="%k", GROUP="audio"
+KERNEL="aload*", NAME="%k", GROUP="audio"
+KERNEL="amidi*", NAME="%k", GROUP="audio"
+KERNEL="amixer*", NAME="%k", GROUP="audio"
+KERNEL="audio*", NAME="%k", GROUP="audio"
+KERNEL="dmfm*", NAME="%k", GROUP="audio"
+KERNEL="dmmidi*", NAME="%k", GROUP="audio"
+KERNEL="dsp*", NAME="%k", GROUP="audio"
+KERNEL="midi*", NAME="%k", GROUP="audio"
+KERNEL="mixer*", NAME="%k", GROUP="audio"
+KERNEL="music", NAME="%k", GROUP="audio"
+KERNEL="sequencer*", NAME="%k", GROUP="audio"
+
+KERNEL="lp[0-9]*", NAME="%k", GROUP="lp"
+KERNEL="parport[0-9]*", NAME="%k", GROUP="lp"
+KERNEL="irlpt[0-9]*", NAME="%k", GROUP="lp"
+KERNEL="usblp[0-9]*", NAME="%k", GROUP="lp"
+
+KERNEL="psaux", NAME="%k", MODE="0644"
+KERNEL="js", NAME="%k", MODE="0644"
+KERNEL="djs", NAME="%k", MODE="0644"
+
+
+KERNEL="fb[0-9]*", NAME="%k", GROUP="video", MODE="0620"
+KERNEL="agpgart", NAME="%k", GROUP="video"
+KERNEL="nvidia*", NAME="%k", GROUP="video"
+
+KERNEL="video[0-9]*", NAME="%k", GROUP="video"
+KERNEL="radio[0-9]*", NAME="%k", GROUP="video"
+KERNEL="vbi[0-9]*", NAME="%k", GROUP="video"
+KERNEL="vtx[0-9]*", NAME="%k", GROUP="video"
+
+
+# Block devices.
+
+KERNEL="fd[0-9]*", NAME="%k", GROUP="floppy"
+
+KERNEL="ram[0-9]*", NAME="%k", GROUP="disk"
+
+KERNEL="hd*", NAME="%k", GROUP="disk"
+
+KERNEL="sd[a-z]", NAME="%k", GROUP="disk"
+KERNEL="sd[a-z][0-9]*", NAME="%k", GROUP="disk"
+KERNEL="sd[a-i][a-z]", NAME="%k", GROUP="disk"
+KERNEL="sd[a-i][a-z][0-9]*", NAME="%k", GROUP="disk"
+KERNEL="s[grt][0-9]*", NAME="%k", GROUP="disk"
+KERNEL="scd[0-9]*", NAME="%k", GROUP="cdrom"
+
+KERNEL="dasd[0-9]*", NAME="%k", GROUP="disk"
+KERNEL="ataraid[0-9]*", NAME="%k", GROUP="disk"
+
+KERNEL="loop[0-9]*", NAME="%k", GROUP="disk"
+KERNEL="md[0-9]*", NAME="%k", GROUP="disk"
+KERNEL="ht[0-9]*", NAME="%k", GROUP="tape"
+KERNEL="nht[0-9]*", NAME="%k", GROUP="tape"
+KERNEL="pt[0-9]*", NAME="%k", GROUP="tape"
+KERNEL="npt[0-9]*", NAME="%k", GROUP="tape"
+KERNEL="st[0-9]*", NAME="%k", GROUP="tape"
+KERNEL="nst[0-9]*", NAME="%k", GROUP="tape"
+
+KERNEL="sgi_fetchop", NAME="%k", MODE="0666"
+KERNEL="iseries/vcd*", NAME="%k", GROUP="disk"
+KERNEL="iseries/vd*", NAME="%k", GROUP="disk"
