--- kdeutils-4.0.1/kwallet/CMakeLists.txt.orig	2008-02-08 21:38:47.000000000 +0100
+++ kdeutils-4.0.1/kwallet/CMakeLists.txt	2008-02-08 21:39:58.000000000 +0100
@@ -19,7 +19,7 @@
    allyourbase.cpp 
    kbetterthankdialogbase.cpp )
 
-qt4_add_dbus_interface(kwalletmanager_SRCS ${DBUS_INTERFACES_INSTALL_DIR}/org.kde.KWallet.xml kwallet_interface)
+qt4_add_dbus_interface(kwalletmanager_SRCS ${KDE4_DBUS_INTERFACES_DIR}/org.kde.KWallet.xml kwallet_interface)
 
 kde4_add_ui_files(kwalletmanager_SRCS walletwidget.ui kbetterthankdialogbase.ui)
 
