--- libopenraw-0.0.9/lib/jfifcontainer.cpp.orig	2016-08-26 01:27:06.523755845 +0000
+++ libopenraw-0.0.9/lib/jfifcontainer.cpp	2016-08-26 01:31:55.559750178 +0000
@@ -145,7 +145,7 @@ namespace OpenRaw {
 		{
 			int ret = 0;
 			if (::setjmp(m_jpegjmp) == 0) {
-				ret = JPEG::jpeg_read_header(&m_cinfo, TRUE);
+				ret = JPEG::jpeg_read_header(&m_cinfo, JPEG::TRUE);
 				//Trace(DEBUG1) << "jpeg_read_header " << ret << "\n";
 				
 				JPEG::jpeg_calc_output_dimensions(&m_cinfo);
@@ -181,7 +181,7 @@ namespace OpenRaw {
 				src->pub.next_input_byte = NULL;
 				src->pub.bytes_in_buffer = 0;
 			}
-			return TRUE;
+			return JPEG::TRUE;
 		}
 
 
