--- linux.vanilla-2.6.9/drivers/scsi/aic7xxx/aic79xx_core.c	2004-10-20 22:32:52.000000000 +0100
+++ linux-2.6.9/drivers/scsi/aic7xxx/aic79xx_core.c	2004-10-24 17:37:01.941748752 +0100
@@ -5270,7 +5270,6 @@
 	default:
 	case 5:
 		ahd_shutdown(ahd);
-		TAILQ_REMOVE(&ahd_tailq, ahd, links);
 		/* FALLTHROUGH */
 	case 4:
 		ahd_dmamap_unload(ahd, ahd->shared_data_dmat,
--- linux.vanilla-2.6.9/drivers/scsi/aic7xxx/aic79xx_osm.c	2004-10-20 23:16:55.000000000 +0100
+++ linux-2.6.9/drivers/scsi/aic7xxx/aic79xx_osm.c	2004-10-24 17:37:01.000000000 +0100
@@ -5090,7 +5090,6 @@
 ahd_linux_exit(void)
 {
 	struct ahd_softc *ahd;
-	u_long l;
 
 	/*
 	 * Shutdown DV threads before going into the SCSI mid-layer.
@@ -5098,12 +5097,11 @@
 	 * kernel so that waiting for our DV threads to exit leads
 	 * to deadlock.
 	 */
-	ahd_list_lock(&l);
 	TAILQ_FOREACH(ahd, &ahd_tailq, links) {
 
 		ahd_linux_kill_dv_thread(ahd);
 	}
-	ahd_list_unlock(&l);
+
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2,5,0)
 	/*
 	 * In 2.4 we have to unregister from the PCI core _after_
--- linux.vanilla-2.6.9/drivers/scsi/aic7xxx/aic79xx_osm_pci.c	2004-10-20 22:32:51.000000000 +0100
+++ linux-2.6.9/drivers/scsi/aic7xxx/aic79xx_osm_pci.c	2004-10-24 17:37:01.000000000 +0100
@@ -92,12 +92,14 @@
 	if (ahd != NULL) {
 		u_long s;
 
+		TAILQ_REMOVE(&ahd_tailq, ahd, links);
+		ahd_list_unlock(&l);
 		ahd_lock(ahd, &s);
 		ahd_intr_enable(ahd, FALSE);
 		ahd_unlock(ahd, &s);
 		ahd_free(ahd);
-	}
-	ahd_list_unlock(&l);
+	} else
+		ahd_list_unlock(&l);
 }
 #endif /* !LINUX_VERSION_CODE < KERNEL_VERSION(2,4,0) */
 
--- linux.vanilla-2.6.9/drivers/scsi/aic7xxx/aic7xxx_core.c	2004-10-20 22:32:51.000000000 +0100
+++ linux-2.6.9/drivers/scsi/aic7xxx/aic7xxx_core.c	2004-10-24 17:37:01.000000000 +0100
@@ -3973,7 +3973,6 @@
 	default:
 	case 5:
 		ahc_shutdown(ahc);
-		TAILQ_REMOVE(&ahc_tailq, ahc, links);
 		/* FALLTHROUGH */
 	case 4:
 		ahc_dmamap_unload(ahc, ahc->shared_data_dmat,
--- linux.vanilla-2.6.9/drivers/scsi/aic7xxx/aic7xxx_osm.c	2004-10-20 23:16:55.000000000 +0100
+++ linux-2.6.9/drivers/scsi/aic7xxx/aic7xxx_osm.c	2004-10-24 17:37:01.000000000 +0100
@@ -5098,7 +5098,6 @@
 ahc_linux_exit(void)
 {
 	struct ahc_softc *ahc;
-	u_long l;
 
 	/*
 	 * Shutdown DV threads before going into the SCSI mid-layer.
@@ -5106,12 +5105,10 @@
 	 * kernel so that waiting for our DV threads to exit leads
 	 * to deadlock.
 	 */
-	ahc_list_lock(&l);
 	TAILQ_FOREACH(ahc, &ahc_tailq, links) {
 
 		ahc_linux_kill_dv_thread(ahc);
 	}
-	ahc_list_unlock(&l);
 
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2,5,0)
 	/*
--- linux.vanilla-2.6.9/drivers/scsi/aic7xxx/aic7xxx_osm_pci.c	2004-10-20 22:32:51.000000000 +0100
+++ linux-2.6.9/drivers/scsi/aic7xxx/aic7xxx_osm_pci.c	2004-10-24 17:37:01.000000000 +0100
@@ -95,12 +95,14 @@
 	if (ahc != NULL) {
 		u_long s;
 
+		TAILQ_REMOVE(&ahc_tailq, ahc, links);
+		ahc_list_unlock(&l);
 		ahc_lock(ahc, &s);
 		ahc_intr_enable(ahc, FALSE);
 		ahc_unlock(ahc, &s);
 		ahc_free(ahc);
-	}
-	ahc_list_unlock(&l);
+	} else
+		ahc_list_unlock(&l);
 }
 #endif /* !LINUX_VERSION_CODE < KERNEL_VERSION(2,4,0) */
 
