Index: linux-2.6.9-ck3/drivers/media/video/cpia.c
===================================================================
--- linux-2.6.9-ck3.orig/drivers/media/video/cpia.c	2004-10-19 08:57:06.000000000 +1000
+++ linux-2.6.9-ck3/drivers/media/video/cpia.c	2004-11-05 23:35:59.578945323 +1100
@@ -4066,22 +4066,13 @@ static int __init cpia_init(void)
 	proc_cpia_create();
 #endif
 
-#ifdef CONFIG_KMOD
-#ifdef CONFIG_VIDEO_CPIA_PP_MODULE
-	request_module("cpia_pp");
-#endif
-
-#ifdef CONFIG_VIDEO_CPIA_USB_MODULE
-	request_module("cpia_usb");
-#endif
-#endif	/* CONFIG_KMOD */
-
 #ifdef CONFIG_VIDEO_CPIA_PP
 	cpia_pp_init();
 #endif
 #ifdef CONFIG_VIDEO_CPIA_USB
 	cpia_usb_init();
 #endif
+
 	return 0;
 }
 
