--- zlib-1.2.3/configure.orig	2006-07-29 14:10:09.000000000 -0300
+++ zlib-1.2.3/configure	2006-07-29 14:42:13.000000000 -0300
@@ -435,7 +435,7 @@ case $CFLAGS in
     fi;;
 esac
 
-rm -f $test.[co] $test $test$shared_ext
+# rm -f $test.[co] $test $test$shared_ext
 
 # udpate Makefile
 sed < Makefile.in "
--- zlib-1.2.3/Makefile.in.orig	2006-07-29 14:42:22.000000000 -0300
+++ zlib-1.2.3/Makefile.in	2006-07-29 14:43:18.000000000 -0300
@@ -79,7 +79,7 @@ match.o: match.S
 
 $(SHAREDLIBV): $(OBJS)
 	$(LDSHARED) -o $@ $(OBJS)
-	rm -f $(SHAREDLIB) $(SHAREDLIBM)
+	# rm -f $(SHAREDLIB) $(SHAREDLIBM)
 	ln -s $@ $(SHAREDLIB)
 	ln -s $@ $(SHAREDLIBM)
 
@@ -96,8 +96,8 @@ install: $(LIBS)
 	-@if [ ! -d $(man3dir)     ]; then mkdir -p $(man3dir); fi
 	cp zlib.h zconf.h $(includedir)
 	chmod 644 $(includedir)/zlib.h $(includedir)/zconf.h
-	cp $(LIBS) $(libdir)
-	cd $(libdir); chmod 755 $(LIBS)
+	cp $(LIBS) $(SHAREDLIBV) $(libdir)
+	cd $(libdir); chmod 755 $(LIBS) $(SHAREDLIBV)
 	-@(cd $(libdir); $(RANLIB) libz.a || true) >/dev/null 2>&1
 	cd $(libdir); if test -f $(SHAREDLIBV); then \
 	  rm -f $(SHAREDLIB) $(SHAREDLIBM); \
