Disables GUI because it includes Wayland headers.

diff -Nur nestopia-1.47.old/Makefile nestopia-1.47/Makefile
--- nestopia-1.47.old/Makefile	2016-01-09 21:34:38.000000000 -0200
+++ nestopia-1.47/Makefile	2016-08-04 00:19:41.789978392 -0300
@@ -46,17 +46,17 @@
 	DEFINES = -DDATADIR=\"$(DATADIR)\"
 	LIBS += -larchive -lepoxy -lGL -lGLU -lao
 	# GTK Stuff - Comment this section to disable GTK+
-	CFLAGS += $(shell pkg-config --cflags gtk+-3.0)
-	LIBS += $(shell pkg-config --libs gtk+-3.0)
-	DEFINES += -D_GTK
-	IOBJS += objs/unix/gtkui/gtkui.o
-	IOBJS += objs/unix/gtkui/gtkui_archive.o
-	IOBJS += objs/unix/gtkui/gtkui_callbacks.o
-	IOBJS += objs/unix/gtkui/gtkui_cheats.o
-	IOBJS += objs/unix/gtkui/gtkui_config.o
-	IOBJS += objs/unix/gtkui/gtkui_dialogs.o
-	OBJDIRS += objs/unix/gtkui
-	WARNINGS += -Wno-deprecated-declarations
+	#CFLAGS += $(shell pkg-config --cflags gtk+-3.0)
+	#LIBS += $(shell pkg-config --libs gtk+-3.0)
+	#DEFINES += -D_GTK
+	#IOBJS += objs/unix/gtkui/gtkui.o
+	#IOBJS += objs/unix/gtkui/gtkui_archive.o
+	#IOBJS += objs/unix/gtkui/gtkui_callbacks.o
+	#IOBJS += objs/unix/gtkui/gtkui_cheats.o
+	#IOBJS += objs/unix/gtkui/gtkui_config.o
+	#IOBJS += objs/unix/gtkui/gtkui_dialogs.o
+	#OBJDIRS += objs/unix/gtkui
+	#WARNINGS += -Wno-deprecated-declarations
 	# end GTK
 endif
 
