--- Mesa-7.0.1/src/glx/x11/glxext.c.orig	2007-08-12 10:45:15.688553249 +0200
+++ Mesa-7.0.1/src/glx/x11/glxext.c	2007-08-12 10:55:49.810243457 +0200
@@ -672,8 +672,11 @@ filter_modes( __GLcontextModes ** server
 	if ( do_delete && (m->visualID != 0) ) {
 	    do_delete = GL_FALSE;
 
-	    fprintf(stderr, "libGL warning: 3D driver claims to not support "
-		    "visual 0x%02x\n", m->visualID);
+	    /* don't warn for this visual (Novell #247471 / X.Org #6689) */
+	    if (m->visualRating != GLX_NON_CONFORMANT_CONFIG) {
+		fprintf(stderr, "libGL warning: 3D driver claims to not "
+			"support visual 0x%02x\n", m->visualID);
+	    }
 	}
 
 	if ( do_delete ) {
