Description: Pkgconfig file

diff -Naur libmad-0.15.1b/mad.pc libmad-0.15.1b.new/mad.pc
--- libmad-0.15.1b/mad.pc	1970-01-01 01:00:00.000000000 +0100
+++ libmad-0.15.1b.new/mad.pc	2008-02-17 21:57:43.000000000 +0100
@@ -0,0 +1,10 @@
+prefix=/usr
+exec_prefix=${prefix}
+libdir=${exec_prefix}/lib
+includedir=${prefix}/include
+
+Name: MAD
+Description: MPEG Audio Decoder
+Version: 0.15.1b
+Libs: -L${libdir} -lmad 
+Cflags: -I${includedir}
