--- a/sigc++/signal.h.orig	2008-02-11 21:11:19.000000000 -0800
+++ b/sigc++/signal.h	2008-02-11 21:15:19.000000000 -0800
@@ -18,7 +18,11 @@
   //Compilers, such as older versions of SUN Forte C++, that do not allow this also often
   //do not allow a typedef to have the same name as a class in the typedef's definition.
   //For Sun Forte CC 5.7 (SUN Workshop 10), comment this out to fix the build.
-  #define SIGC_TYPEDEF_REDEFINE_ALLOWED 1
+  //
+  // Debian-specific patch: gcc-4.3 does not allow this either, and the
+  // comments below say this is a backwards-compatibility typedef, so
+  // just don't enable this feature at all.
+  //#define SIGC_TYPEDEF_REDEFINE_ALLOWED 1
 #endif
 
 namespace sigc {
