Unionfs: update lower mnts on rmdir with copyup

Signed-off-by: Erez Zadok <ezk@cs.sunysb.edu>

--- linux-2.6.24.4/fs/unionfs/unlink.c.orig	2008-03-30 19:22:43.000000000 -0700
+++ linux-2.6.24.4/fs/unionfs/unlink.c	2008-03-30 19:22:50.000000000 -0700
@@ -280,6 +280,8 @@ out:
 		    !unionfs_lower_inode_idx(inode, dend))
 			ibstart(inode) = ibend(inode) = -1;
 		d_drop(dentry);
+		/* update our lower vfsmnts, in case a copyup took place */
+		unionfs_postcopyup_setmnt(dentry);
 	}
 
 	if (namelist)
