--- webkitgtk-1.11.92/Source/JavaScriptCore/API/JSStringRef.cpp.orig	2012-10-30 05:47:55.000000000 -0200
+++ webkitgtk-1.11.92/Source/JavaScriptCore/API/JSStringRef.cpp	2019-01-14 20:57:15.981530644 -0200
@@ -36,7 +36,7 @@ using namespace WTF::Unicode;
 JSStringRef JSStringCreateWithCharacters(const JSChar* chars, size_t numChars)
 {
     initializeThreading();
-    return OpaqueJSString::create(chars, numChars).leakRef();
+    return OpaqueJSString::create((const UChar*)chars, numChars).leakRef();
 }
 
 JSStringRef JSStringCreateWithUTF8CString(const char* string)
@@ -77,7 +77,7 @@ size_t JSStringGetLength(JSStringRef str
 
 const JSChar* JSStringGetCharactersPtr(JSStringRef string)
 {
-    return string->characters();
+    return (const JSChar *)string->characters();
 }
 
 size_t JSStringGetMaximumUTF8CStringSize(JSStringRef string)
