--- lua-fs-0.1_orig/libfs.c	2005-11-07 04:30:32.000000000 +0100
+++ lua-fs-0.1/libfs.c	2007-04-14 09:46:49.000000000 +0200
@@ -41,7 +41,7 @@
 #include <pwd.h>
 
  
-#define MYNAME		"libfs"
+#define MYNAME		"fs"
 #define VERSION	        "$Id: libfs.c,v 1.11 2005/11/07 03:33:29 steve Exp $"
 
 
@@ -584,7 +584,7 @@
  * @param L The lua intepreter object.
  * @return 1 on success, 0 on failure.
  */
-LUALIB_API int luaopen_libfs (lua_State *L)
+LUALIB_API int luaopen_fs (lua_State *L)
 {
     /* Version number from CVS marker. */
     char *version = getVersion();
