Fix spurious "unsafe include" errors
diff -u asciidoc-8.2.7-orig/asciidoc.py asciidoc-8.2.7/asciidoc.py
--- asciidoc-8.2.7-orig/asciidoc.py     2008-07-03 11:15:11.000000000 +1200
+++ asciidoc-8.2.7/asciidoc.py  2008-11-27 10:43:24.000000000 +1300
@@ -125,7 +125,7 @@
     else:
         assert os.path.isdir(directory)
         directory = os.path.abspath(directory)
-    fname = os.path.realpath(fname)
+    fname = os.path.abspath(fname)
     return os.path.commonprefix((directory, fname)) == directory

 def safe():

