--- bakefile-0.2.0/admin/python-devel.m4.orig	2006-08-11 15:18:10.000000000 -0300
+++ bakefile-0.2.0/admin/python-devel.m4	2006-08-11 15:18:40.000000000 -0300
@@ -26,7 +26,7 @@ AC_DEFUN([BKL_PYTHON_DEVEL],[
     fi
     # Apple's Python not found, try standard search:
     if test -z "$python_ldflags" ; then
-        python_path=${PYTHON%/bin*}
+        python_path=${PYTHON_PATH_AT_GOBOPROGRAMS}
         for i in "$python_path/lib/python$PYTHON_VERSION/config/" "$python_path/lib/python$PYTHON_VERSION/" "$python_path/lib/python/config/" "$python_path/lib/python/" "$python_path/" ; do
             python_path=`find $i -type f -name "libpython$PYTHON_VERSION.*" -print | head -n 1 2>/dev/null`
             if test -n "$python_path" ; then
