diff -Naur kdelibs-3.2.2/kdecore/Makefile.am kdelibs-3.2.2-GOBO/kdecore/Makefile.am
--- kdelibs-3.2.2/kdecore/Makefile.am	2003-11-30 04:46:42.000000000 -0500
+++ kdelibs-3.2.2-GOBO/kdecore/Makefile.am	2004-05-07 22:12:35.754249104 -0400
@@ -163,9 +163,9 @@
 
 # kgrantpty needs to be installed setuid root
 install-exec-hook: 
-	@(chown root $(DESTDIR)$(bindir)/kgrantpty && chmod 4755 $(DESTDIR)$(bindir)/kgrantpty) || echo "Please make kgrantpty setuid root" >&2
+	@(chown gobo $(DESTDIR)$(bindir)/kgrantpty && chmod 4755 $(DESTDIR)$(bindir)/kgrantpty) || echo "Please make kgrantpty setuid gobo" >&2
 	@echo ""
-	@echo "kgrantpty is by default installed with a set SETUID root bit!"
+	@echo "kgrantpty is by default installed with a set SETUID gobo bit!"
 	@echo "This is needed for konsole, etc. to ensure that they can't be eavesdroped."
 	@echo ""
 
diff -Naur kdelibs-3.2.2/kdecore/Makefile.in kdelibs-3.2.2-GOBO/kdecore/Makefile.in
--- kdelibs-3.2.2/kdecore/Makefile.in	2004-04-05 11:12:26.000000000 -0400
+++ kdelibs-3.2.2-GOBO/kdecore/Makefile.in	2004-05-07 22:12:35.760248192 -0400
@@ -1687,9 +1687,9 @@
 
 # kgrantpty needs to be installed setuid root
 install-exec-hook: 
-	@(chown root $(DESTDIR)$(bindir)/kgrantpty && chmod 4755 $(DESTDIR)$(bindir)/kgrantpty) || echo "Please make kgrantpty setuid root" >&2
+	@(chown gobo $(DESTDIR)$(bindir)/kgrantpty && chmod 4755 $(DESTDIR)$(bindir)/kgrantpty) || echo "Please make kgrantpty setuid gobo" >&2
 	@echo ""
-	@echo "kgrantpty is by default installed with a set SETUID root bit!"
+	@echo "kgrantpty is by default installed with a set SETUID gobo bit!"
 	@echo "This is needed for konsole, etc. to ensure that they can't be eavesdroped."
 	@echo ""
 apidox-am-yes:
diff -Naur kdelibs-3.2.2/kio/misc/kpac/Makefile.am kdelibs-3.2.2-GOBO/kio/misc/kpac/Makefile.am
--- kdelibs-3.2.2/kio/misc/kpac/Makefile.am	2004-01-17 07:53:18.000000000 -0500
+++ kdelibs-3.2.2-GOBO/kio/misc/kpac/Makefile.am	2004-05-07 22:13:20.825397248 -0400
@@ -24,4 +24,4 @@
 proxyscoutdata_DATA = eventsrc
 
 install-exec-local:
-	@(chown root $(DESTDIR)$(bindir)/kpac_dhcp_helper && chmod 4755 $(DESTDIR)$(bindir)/kpac_dhcp_helper) || echo "Please make kpac_dhcp_helper setuid root"
+	@(chown gobo $(DESTDIR)$(bindir)/kpac_dhcp_helper && chmod 4755 $(DESTDIR)$(bindir)/kpac_dhcp_helper) || echo "Please make kpac_dhcp_helper setuid gobo"
diff -Naur kdelibs-3.2.2/kio/misc/kpac/Makefile.in kdelibs-3.2.2-GOBO/kio/misc/kpac/Makefile.in
--- kdelibs-3.2.2/kio/misc/kpac/Makefile.in	2004-04-05 11:12:31.000000000 -0400
+++ kdelibs-3.2.2-GOBO/kio/misc/kpac/Makefile.in	2004-05-07 22:13:20.825397248 -0400
@@ -917,7 +917,7 @@
 
 
 install-exec-local:
-	@(chown root $(DESTDIR)$(bindir)/kpac_dhcp_helper && chmod 4755 $(DESTDIR)$(bindir)/kpac_dhcp_helper) || echo "Please make kpac_dhcp_helper setuid root"
+	@(chown gobo $(DESTDIR)$(bindir)/kpac_dhcp_helper && chmod 4755 $(DESTDIR)$(bindir)/kpac_dhcp_helper) || echo "Please make kpac_dhcp_helper setuid gobo"
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
