--- openquicktime-1.0-src/audioplugin/MP3/audioencoder/lame3.70/quantize-pvt.c.orig	2004-11-23 02:10:27.000000000 -0200
+++ openquicktime-1.0-src/audioplugin/MP3/audioencoder/lame3.70/quantize-pvt.c	2004-11-23 02:10:53.000000000 -0200
@@ -1007,18 +1007,18 @@
         "dec %4\n\t"
 
         "movl %5, %%eax\n\t"
-        "movl 4+%5, %%ebx\n\t"
+        "movl 4+%5, %%ecx\n\t"
         "fxch %%st(1)\n\t"
         "fadd" F8type " (%2,%%eax," F8size ")\n\t"
         "fxch %%st(3)\n\t"
-        "fadd" F8type " (%2,%%ebx," F8size ")\n\t"
+        "fadd" F8type " (%2,%%ecx," F8size ")\n\t"
 
         "movl 8+%5, %%eax\n\t"
         "movl 12+%5, %%ebx\n\t"
         "fxch %%st(2)\n\t"
         "fadd" F8type " (%2,%%eax," F8size ")\n\t"
         "fxch %%st(1)\n\t"
-        "fadd" F8type " (%2,%%ebx," F8size ")\n\t"
+        "fadd" F8type " (%2,%%ecx," F8size ")\n\t"
 
         "fxch %%st(3)\n\t"
         "fistpl -16(%3)\n\t"
@@ -1030,7 +1030,7 @@
         "jnz loop1\n\n"
         : /* no outputs */
         : "t" (istep), "r" (xr), "r" (adj43asm), "r" (ix), "r" (576 / 4), "m" (rx)
-        : "%eax", "%ebx", "memory", "cc"
+        : "%eax", "%ecx", "memory", "cc"
       );
   }
 #elif defined (USE_MSC_ASM)
