Adds a PREFIX variable and creates the target install directory.

diff -Nur alsaequal.old/Makefile alsaequal/Makefile
--- alsaequal.old/Makefile	2010-03-05 14:30:39.000000000 -0300
+++ alsaequal/Makefile	2017-01-17 02:35:35.322951502 -0200
@@ -1,4 +1,6 @@
 
+PREFIX=/usr
+
 # Quiet (set to @ for a quite compile)
 Q	?= @
 #Q	?=
@@ -45,11 +47,12 @@
 
 install: all
 	@echo Installing...
-	$(Q)install -m 644 $(SND_PCM_BIN) ${DESTDIR}/usr/lib/alsa-lib/
-	$(Q)install -m 644 $(SND_CTL_BIN) ${DESTDIR}/usr/lib/alsa-lib/
+	$(Q)mkdir -p "${DESTDIR}${PREFIX}/lib/alsa-lib/"
+	$(Q)install -m 644 $(SND_PCM_BIN) "${DESTDIR}${PREFIX}/lib/alsa-lib/"
+	$(Q)install -m 644 $(SND_CTL_BIN) "${DESTDIR}${PREFIX}/lib/alsa-lib/"
 
 uninstall:
 	@echo Un-installing...
-	$(Q)rm ${DESTDIR}/usr/lib/alsa-lib/$(SND_PCM_BIN)
-	$(Q)rm ${DESTDIR}/usr/lib/alsa-lib/$(SND_CTL_BIN)
+	$(Q)rm "${DESTDIR}${PREFIX}/lib/alsa-lib/$(SND_PCM_BIN)"
+	$(Q)rm "${DESTDIR}${PREFIX}/lib/alsa-lib/$(SND_CTL_BIN)"
 	
