--- grub-2.02~beta3/grub-core/osdep/linux/getroot.c.orig	2016-10-07 16:51:05.781203692 +0000
+++ grub-2.02~beta3/grub-core/osdep/linux/getroot.c	2016-10-07 17:19:23.581170401 +0000
@@ -370,6 +370,19 @@ get_btrfs_fs_prefix (const char *mount_p
   return NULL;
 }
 
+void
+translate_from_gobolinux (char *device)
+{
+  const char *gobodev = "/System/Kernel/Devices";
+  char *sep = strstr(device, gobodev);
+  if (sep == device)
+  {
+    char copy[strlen(device)+1];
+    int n = sprintf(copy, "/dev%s", &device[strlen(gobodev)]);
+    copy[n] = '\0';
+    strcpy(device, copy);
+  }
+}
 
 char **
 grub_find_root_devices_from_mountinfo (const char *dir, char **relroot)
@@ -426,6 +439,7 @@ grub_find_root_devices_from_mountinfo (c
       if (sscanf (sep, "%s %s", entry.fstype, entry.device) != 2)
 	continue;
 
+      translate_from_gobolinux (entry.device);
       unescape (entry.device);
 
       /* Using the mount IDs, find out where this fits in the list of
