--- SFCGAL-1.3.0/CMakeLists.txt.orig	2017-05-04 18:39:13.820691379 -0300
+++ SFCGAL-1.3.0/CMakeLists.txt	2017-05-04 19:00:33.613715388 -0300
@@ -80,6 +80,15 @@ if((${Boost_MAJOR_VERSION} EQUAL 1) AND
   include_directories( patches/boost-1.60.0 )
 endif()
 
+find_package( osg )
+find_package( osgDB )
+if ( ${OSG_FOUND} AND ${OSGDB_FOUND} )
+	message( STATUS "OSG_INCLUDE_DIR = ${OSG_INCLUDE_DIR}" )
+	message( STATUS "OSG_LIBRARY = ${OSG_LIBRARY}" )
+    message( STATUS "OSGDB_INCLUDE_DIR = ${OSGDB_INCLUDE_DIR}" )
+    message( STATUS "OSGDB_LIBRARY = ${OSGDB_LIBRARY}" )
+endif()
+
 #-- GMP (facultative)  -------------------------------------
 find_package( GMP )
 if( ${GMP_FOUND} )
@@ -132,6 +141,8 @@ include_directories( SYSTEM
 	${Boost_INCLUDE_DIRS}	
 	${MPFR_INCLUDE_DIRS}	
 	${GMP_INCLUDE_DIRS}	
+	${OSG_INCLUDE_DIR}	
+    ${OSGDB_INCLUDE_DIR}	
 )
 link_directories(
 	${CGAL_LIBRARY_DIRS}
--- SFCGAL-1.3.0/src/CMakeLists.txt.orig	2017-05-04 18:50:01.472703529 -0300
+++ SFCGAL-1.3.0/src/CMakeLists.txt	2017-05-04 19:01:00.897715900 -0300
@@ -47,7 +47,8 @@ if( ${SFCGAL_WITH_GMP} )
 endif( ${SFCGAL_WITH_GMP} )
 
 if( ${SFCGAL_WITH_OSG} )
-	 target_link_libraries( SFCGAL ${OPENSCENEGRAPH_LIBRARIES} )
+    message( STATUS "linking target with OSG library" )
+    target_link_libraries( SFCGAL ${OSG_LIBRARY} ${OSGDB_LIBRARY} )
 endif()
 target_link_libraries( SFCGAL ${Boost_LIBRARIES} )
 
