Added install to gadgets
--- dzen-master//Makefile    2013-09-23 06:57:39.000000000 -0300
+++ dzen-master//Makefile    2013-09-23 06:57:39.000000000 -0300
@@ -6,7 +6,10 @@
 SRC = draw.c main.c util.c action.c
 OBJ = ${SRC:.c=.o}
 
-all: options dzen2
+all: gadgets options dzen2
+
+gadgets:
+	@make -C gadgets
 
 options:
 	@echo dzen2 build options:
@@ -48,9 +51,11 @@
 	@mkdir -p ${DESTDIR}${PREFIX}/bin
 	@cp -f dzen2 ${DESTDIR}${PREFIX}/bin
 	@chmod 755 ${DESTDIR}${PREFIX}/bin/dzen2
+	@cp -f gadgets/{dbar,gcpubar,gdbar,textwidth} ${DESTDIR}${PREFIX}/bin
+	@chmod 755 ${DESTDIR}${PREFIX}/bin/{dbar,gcpubar,gdbar,textwidth}
 
 uninstall:
 	@echo removing executable file from ${DESTDIR}${PREFIX}/bin
 	@rm -f ${DESTDIR}${PREFIX}/bin/dzen2
 
-.PHONY: all options clean dist install uninstall
+.PHONY: all gadgets options clean dist install uninstall
