diff -Naur iproute2-2.6.25/Makefile iproute2-2.6.25.new/Makefile
--- iproute2-2.6.25/Makefile	2008-05-07 09:30:28.000000000 +0000
+++ iproute2-2.6.25.new/Makefile	2008-05-07 09:30:47.000000000 +0000
@@ -1,5 +1,6 @@
 DESTDIR=
 SBINDIR=/usr/sbin
+LIBDIR=/usr/lib
 CONFDIR=/etc/iproute2
 DOCDIR=/usr/share/doc/iproute2
 MANDIR=/usr/share/man
diff -Naur iproute2-2.6.25/netem/Makefile iproute2-2.6.25.new/netem/Makefile
--- iproute2-2.6.25/netem/Makefile	2008-05-07 09:30:28.000000000 +0000
+++ iproute2-2.6.25.new/netem/Makefile	2008-05-07 09:32:03.000000000 +0000
@@ -19,9 +19,9 @@
 	$(HOSTCC) $(CCOPTS) -I../include -o $@ $@.c -lm
 
 install: all
-	mkdir -p $(DESTDIR)/usr/lib/tc
+	mkdir -p $(DESTDIR)$(LIBDIR)/tc
 	for i in $(DISTDATA); \
-	do install -m 755 $$i $(DESTDIR)/usr/lib/tc; \
+	do install -m 755 $$i $(DESTDIR)$(LIBDIR)/tc; \
 	done
 
 clean:
diff -Naur iproute2-2.6.25/tc/Makefile iproute2-2.6.25.new/tc/Makefile
--- iproute2-2.6.25/tc/Makefile	2008-05-07 09:30:28.000000000 +0000
+++ iproute2-2.6.25.new/tc/Makefile	2008-05-07 09:31:32.000000000 +0000
@@ -72,10 +72,10 @@
 	$(AR) rcs $@ $(TCLIB)
 
 install: all
-	mkdir -p $(DESTDIR)/usr/lib/tc
+	mkdir -p $(DESTDIR)$(LIBDIR)/tc
 	install -m 0755 tc $(DESTDIR)$(SBINDIR)
 	for i in $(TCSO); \
-	do install -m 755 $$i $(DESTDIR)/usr/lib/tc; \
+	do install -m 755 $$i $(DESTDIR)$(LIBDIR)/tc; \
 	done
 
 clean:
