diff -ruN webkitgtk-2.4.11.orig/Source/WebKit2/Shared/API/c/WKString.cpp webkitgtk-2.4.11/Source/WebKit2/Shared/API/c/WKString.cpp
--- webkitgtk-2.4.11.orig/Source/WebKit2/Shared/API/c/WKString.cpp	2016-04-10 08:48:37.000000000 +0200
+++ webkitgtk-2.4.11/Source/WebKit2/Shared/API/c/WKString.cpp	2017-06-11 19:31:59.544807847 +0200
@@ -55,7 +55,7 @@
 size_t WKStringGetCharacters(WKStringRef stringRef, WKChar* buffer, size_t bufferLength)
 {
     COMPILE_ASSERT(sizeof(WKChar) == sizeof(UChar), WKStringGetCharacters_sizeof_WKChar_matches_UChar);
-    return (toImpl(stringRef)->getCharacters(static_cast<UChar*>(buffer), bufferLength));
+    return (toImpl(stringRef)->getCharacters(reinterpret_cast<UChar*>(buffer), bufferLength));
 }
 
 size_t WKStringGetMaximumUTF8CStringSize(WKStringRef stringRef)
