--- iproute2-2.6.23/netem/Makefile.orig	2008-03-09 17:34:31.000000000 +0100
+++ iproute2-2.6.23/netem/Makefile	2008-03-09 17:35:00.000000000 +0100
@@ -19,9 +19,9 @@
 	$(HOSTCC) $(CCOPTS) -I../include -o $@ $@.c -lm
 
 install: all
-	mkdir -p $(DESTDIR)/usr/lib/tc
+	mkdir -p $(DESTDIR)$(LIBDIR)/tc
 	for i in $(DISTDATA); \
-	do install -m 755 $$i $(DESTDIR)/usr/lib/tc; \
+	do install -m 755 $$i $(DESTDIR)$(LIBDIR)/tc; \
 	done
 
 clean:
