diff -Naur rlocate-0.5.1.old/src/rlocate-scripts/Makefile.am rlocate-0.5.1/src/rlocate-scripts/Makefile.am
--- rlocate-0.5.1.old/src/rlocate-scripts/Makefile.am	2006-05-07 15:51:50.000000000 +0200
+++ rlocate-0.5.1/src/rlocate-scripts/Makefile.am	2006-07-16 22:38:28.000000000 +0200
@@ -18,9 +18,5 @@
 	rm $(srcdir)/rlocate-checkpoint.temp
 
 install-exec-hook:
-	-/usr/sbin/groupadd $(rlocate_group)
 	chown $(root_uid):$(rlocate_group) $(DESTDIR)$(bindir)/rlocate-checkpoint
 	chmod 2755 $(DESTDIR)$(bindir)/rlocate-checkpoint
-
-uninstall-hook:
-	-/usr/sbin/groupdel $(rlocate_group)
