--- NSS/mozilla/security/coreconf/Linux.mk.orig	2005-09-16 17:09:23.000000000 +0000
+++ NSS/mozilla/security/coreconf/Linux.mk	2006-02-09 05:27:36.000000000 +0000
@@ -37,6 +37,8 @@
 
 include $(CORE_DEPTH)/coreconf/UNIX.mk
 
+NSS_VERSION_MINOR=12
+
 #
 # The default implementation strategy for Linux is now pthreads
 #
--- NSS/mozilla/security/coreconf/Linux2.6.mk.orig	2005-09-16 17:09:23.000000000 +0000
+++ NSS/mozilla/security/coreconf/Linux2.6.mk	2006-02-09 05:38:16.000000000 +0000
@@ -38,8 +38,7 @@
 include $(CORE_DEPTH)/coreconf/Linux.mk
 
 OS_REL_CFLAGS   += -DLINUX2_1
-MKSHLIB         = $(CC) $(DSO_LDOPTS) -Wl,-soname -Wl,$(@:$(OBJDIR)/%.so=%.so)
-
+MKSHLIB         = $(CC) $(DSO_LDOPTS) -Wl,-soname -Wl,$(@:$(OBJDIR)/%.so=%.so).$(NSS_VERSION_MINOR)
 ifdef MAPFILE
 	MKSHLIB += -Wl,--version-script,$(MAPFILE)
 endif
