usermod's --append option shouldn't require an option.

Signed-off-by: Lucas C. Villa Real <lucasvr@gobolinux.org>

--- shadow-4.0.18.1/src/usermod.c.orig	2008-01-06 18:07:30.000000000 -0800
+++ shadow-4.0.18.1/src/usermod.c	2008-01-06 18:07:32.000000000 -0800
@@ -912,7 +912,7 @@ static void process_flags (int argc, cha
 		 */
 		int c;
 		static struct option long_options[] = {
-			{"append", required_argument, NULL, 'a'},
+			{"append", no_argument, NULL, 'a'},
 			{"comment", required_argument, NULL, 'c'},
 			{"home", required_argument, NULL, 'd'},
 			{"expiredate", required_argument, NULL, 'e'},
