diff -Naur xc-orig/config/cf/host.def xc/config/cf/host.def
--- xc-orig/config/cf/host.def	2004-04-06 21:16:51.000000000 -0300
+++ xc/config/cf/host.def	2004-05-14 11:37:08.000000000 -0300
@@ -0,0 +1,16 @@
+#define ProjectRoot /Programs/Xorg/6.7.0
+#define EtcX11Directory /Programs/Xorg/Settings/X11
+#define FontconfigFontsConfDir /Programs/Fontconfig/Settings/fonts
+#define Build75DpiFonts 	NO
+#define Build100DpiFonts 	NO
+#define BuildSpeedoFonts 	NO
+#define BuildType1Fonts 	NO
+#define BuildCIDFonts 		NO
+#define BuildTrueTypeFonts 	NO
+#define BuildCyrillicFonts 	NO
+#define HasFontconfig		YES
+#define HasFreetype2		YES
+#define HasExpat		YES
+#define HasLibpng		YES
+#define HasZlib			YES
+
