diff -Naur comix-1.3.8/kwin/Makefile.am comix-1.3.8.new/kwin/Makefile.am
--- comix-1.3.8/kwin/Makefile.am	2006-04-28 11:04:03.000000000 +0200
+++ comix-1.3.8.new/kwin/Makefile.am	2007-11-17 14:50:16.000000000 +0100
@@ -14,7 +14,7 @@
 kde_module_LTLIBRARIES = kwin3_comix.la
 
 kwin3_comix_la_SOURCES = comix.cpp comixClient.cpp comixCircle.cpp comixButton.cpp 
-kwin3_comix_la_LIBADD = $(kde_libraries)/libkdecorations.la
+kwin3_comix_la_LIBADD = -lkdecorations
 kwin3_comix_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -lkdecore -module
 kwin3_comix_la_METASOURCES = AUTO
 
diff -Naur comix-1.3.8/kwin/Makefile.in comix-1.3.8.new/kwin/Makefile.in
--- comix-1.3.8/kwin/Makefile.in	2006-04-28 12:29:00.000000000 +0200
+++ comix-1.3.8.new/kwin/Makefile.in	2007-11-18 13:38:02.000000000 +0100
@@ -60,7 +60,8 @@
 am__installdirs = "$(DESTDIR)$(kde_moduledir)" "$(DESTDIR)$(kwindir)"
 kde_moduleLTLIBRARIES_INSTALL = $(INSTALL)
 LTLIBRARIES = $(kde_module_LTLIBRARIES)
-kwin3_comix_la_DEPENDENCIES = $(kde_libraries)/libkdecorations.la
+am__DEPENDENCIES_1 =
+kwin3_comix_la_DEPENDENCIES = $(am__DEPENDENCIES_1)
 am_kwin3_comix_la_OBJECTS = comix.lo comixClient.lo comixCircle.lo \
 	comixButton.lo
 #>- kwin3_comix_la_OBJECTS = $(am_kwin3_comix_la_OBJECTS)
@@ -346,7 +347,7 @@
 noinst_HEADERS = comix.h comixClient.h comixCircle.h comixButton.h 
 kde_module_LTLIBRARIES = kwin3_comix.la
 kwin3_comix_la_SOURCES = comix.cpp comixClient.cpp comixCircle.cpp comixButton.cpp 
-kwin3_comix_la_LIBADD = $(kde_libraries)/libkdecorations.la
+kwin3_comix_la_LIBADD = -lkdecorations
 kwin3_comix_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -lkdecore -module
 #>- kwin3_comix_la_METASOURCES = AUTO
 DISTCLEANFILES = $(kwin3_comix_la_METASOURCES)
