Fixes the build in 2.6.24.3 kernels.

Signed-off-by: Lucas C. Villa Real <lucasvr@gobolinux.org>

--- linux-2.6.24.3/fs/squashfs/inode.c.orig	2008-03-10 03:32:47.000000000 -0700
+++ linux-2.6.24.3/fs/squashfs/inode.c	2008-03-10 03:38:15.000000000 -0700
@@ -31,6 +31,7 @@
 #include <linux/vfs.h>
 #include <linux/vmalloc.h>
 #include <linux/smp_lock.h>
+#include <linux/exportfs.h>
 
 #include "squashfs.h"
 
@@ -2159,10 +2160,9 @@ static void squashfs_destroy_inode(struc
 }
 
 
-static void init_once(void * foo, struct kmem_cache * cachep, unsigned long flags)
+static void init_once(struct kmem_cache * cachep, void *foo)
 {
 	struct squashfs_inode_info *ei = foo;
-
 	inode_init_once(&ei->vfs_inode);
 }
  
@@ -2171,7 +2171,7 @@ static int __init init_inodecache(void)
 {
 	squashfs_inode_cachep = kmem_cache_create("squashfs_inode_cache",
 	    sizeof(struct squashfs_inode_info), 0,
-		SLAB_HWCACHE_ALIGN|SLAB_RECLAIM_ACCOUNT, init_once, NULL);
+		SLAB_HWCACHE_ALIGN|SLAB_RECLAIM_ACCOUNT, init_once);
 	if (squashfs_inode_cachep == NULL)
 		return -ENOMEM;
 	return 0;
