# fix: ffmpeg api change
diff -Nur opencv-1.0.0/otherlibs/highgui/cvcap_ffmpeg.cpp opencv-1.0.0-fabio.m/otherlibs/highgui/cvcap_ffmpeg.cpp
--- opencv-1.0.0/otherlibs/highgui/cvcap_ffmpeg.cpp	2006-07-24 19:27:39.000000000 -0300
+++ opencv-1.0.0-fabio.m/otherlibs/highgui/cvcap_ffmpeg.cpp	2008-09-23 19:03:12.000000000 -0300
@@ -781,7 +781,7 @@
 
 	if (!(mywriter->fmt->flags & AVFMT_NOFILE)) {
 		/* close the output file */
-		url_fclose(&mywriter->oc->pb);
+		url_fclose(mywriter->oc->pb);
 	}
 
 	/* free the stream */
