--- dpkg-1.18.18/lib/dpkg/Makefile.in.orig	2016-12-24 19:58:42.061773123 -0200
+++ dpkg-1.18.18/lib/dpkg/Makefile.in	2016-12-24 19:58:52.029773310 -0200
@@ -91,11 +91,11 @@ POST_UNINSTALL = :
 build_triplet = @build@
 host_triplet = @host@
 
-@BUILD_SHARED_TRUE@am__append_2 = \
-@BUILD_SHARED_TRUE@	$(LIBINTL) \
-@BUILD_SHARED_TRUE@	$(Z_LIBS) \
-@BUILD_SHARED_TRUE@	$(LZMA_LIBS) \
-@BUILD_SHARED_TRUE@	$(BZ2_LIBS)
+am__append_2 = \
+	$(LIBINTL) \
+	$(Z_LIBS) \
+	$(LZMA_LIBS) \
+	$(BZ2_LIBS)
 
 subdir = lib/dpkg
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
