--- iproute2-2.6.23/tc/Makefile.orig	2007-10-16 23:27:42.000000000 +0200
+++ iproute2-2.6.23/tc/Makefile	2008-03-09 17:19:31.000000000 +0100
@@ -70,10 +70,10 @@
 	$(AR) rcs $@ $(TCLIB)
 
 install: all
-	mkdir -p $(DESTDIR)/usr/lib/tc
+	mkdir -p $(DESTDIR)$(LIBDIR)/tc
 	install -m 0755 -s tc $(DESTDIR)$(SBINDIR)
 	for i in $(TCSO); \
-	do install -m 755 -s $$i $(DESTDIR)/usr/lib/tc; \
+	do install -m 755 -s $$i $(DESTDIR)$(LIBDIR)/tc; \
 	done
 
 clean:
