--- clisp-2.41_orig/configure	2006-09-29 03:01:48.000000000 +0200
+++ clisp-2.41/configure	2007-01-30 22:56:20.000000000 +0100
@@ -169,15 +169,13 @@
     case "$arg" in
 
       --build | --buil | --bui | --bu)
-        do_config=1
         do_build=1 ;;
 
       --config | --confi | --conf | --con | --co | --c)
         do_config=1 ;;
 
       --install | --instal | --insta | --inst | --ins)
-        do_config=1
-        do_build=1
+      	do_build=1
         do_install=1 ;;
 
       --edit-config | --edit-confi | --edit-conf | --edit-con | --edit-co | \
@@ -713,31 +711,8 @@
   fi
 fi
 
-if test -z "$do_build"; then
-
-  echo
-  echo "To continue building CLISP, the following commands are recommended"
-  echo "  (cf. unix/INSTALL step 4):"
-  if test "$DIRNAME" != "."; then
-    echo "    cd $DIRNAME"
-  fi
-  echo "    ./makemake $makemake_args > Makefile"
-  echo "    make config.lisp"
-  echo "    ${EDITOR:-vi} config.lisp"
-  stacksizelimit=`ulimit -s 2>/dev/null || :` # cygwin /bin/sh ulimit is broken
-  if test -z "$stacksizelimit" || { test "$stacksizelimit" != unlimited && test "$stacksizelimit" -lt 8192; }; then
-    cat <<EOF
-# The default stack size on your platform is insufficient
-# and must be increased to at least 8192.  You must do either
-#    'ulimit -s 8192' (for Bourne shell derivatives, e.g., bash and zsh) or
-#    'limit stacksize 8192' (for C shell derivarives, e.g., tcsh)
-EOF
-  fi
-  echo "    make"
-  echo "    make check"
-
-else
-
+# Creates the makefile with makemake
+if [ ! -z $do_config ]; then
   cd $ABS_DIRNAME
   echo "./makemake $makemake_args > Makefile"
   ./makemake $makemake_args > Makefile
@@ -748,14 +723,30 @@
   # ulimit: just to make sure, should not hurt
   #         cygwin /bin/sh ulimit is broken
   set +e; ulimit -s 8192 2>/dev/null; set -e;
-  make
-  make check
   cd ..
+fi
 
+if test -z "$do_build"; then
+if test -z "$do_install"; then
+  stacksizelimit=`ulimit -s 2>/dev/null || :` # cygwin /bin/sh ulimit is broken
+  if test -z "$stacksizelimit" || { test "$stacksizelimit" != unlimited && test "$stacksizelimit" -lt 8192; }; then
+    cat <<EOF
+# The default stack size on your platform is insufficient
+# and must be increased to at least 8192.  You must do either
+#    'ulimit -s 8192' (for Bourne shell derivatives, e.g., bash and zsh) or
+#    'limit stacksize 8192' (for C shell derivarives, e.g., tcsh)
+EOF
+  fi
+fi
+else
+	# Compile
+	cd $ABS_DIRNAME
+	make
+	cd ..
 fi
 
 if test -n "$do_install"; then
-
+  # Install
   cd $ABS_DIRNAME
   make install
   cd ..
