--- tapioca-qt-0.14.1/CMakeLists.txt.orig	2008-05-13 01:42:26.000000000 -0300
+++ tapioca-qt-0.14.1/CMakeLists.txt	2008-05-13 01:42:32.000000000 -0300
@@ -51,7 +51,7 @@ install(FILES ${CMAKE_BINARY_DIR}/QtTapi
 
 add_subdirectory(src)
 add_subdirectory(include)
-add_subdirectory(tests)
+#add_subdirectory(tests)
 
 # If the cmake version includes cpack, use it
 IF(EXISTS "${CMAKE_ROOT}/Modules/CPack.cmake")
