--- /dev/null	2020-03-10 09:12:00.753908607 -0300
+++ SyncTeX-git/Makefile	2020-03-10 15:54:54.766811193 -0300
@@ -0,0 +1,29 @@
+.PHONY: clean
+
+CC      = gcc
+HEADERS = synctex_parser.h synctex_parser_utils.h
+SOURCES = $(patsubst %.h,%.c, $(HEADERS))
+OBJS    = $(patsubst %.h,%.o, $(HEADERS))
+
+CFLAGS  = -I. -Wall -O3 -fPIC
+LDFLAGS = -shared -fPIC -Wl,-soname,libsynctex.so.1
+
+TARGET = libsynctex.so.1
+
+all: $(TARGET)
+
+install: $(TARGET)
+	@mkdir -p $(PREFIX)/lib/pkgconfig $(PREFIX)/include
+	@cp -v synctex.pc $(PREFIX)/lib/pkgconfig
+	@cp -v *.h $(PREFIX)/include
+	@cp -v $(TARGET) $(PREFIX)/lib
+	@ln -sf $(TARGET) $(PREFIX)/lib/libsynctex.so
+
+$(TARGET): $(OBJS)
+	$(CC) $^ -o $@ ${LDFLAGS}
+
+$(OBJS): $(HEADERS) $(SOURCES)
+
+
+clean:
+	rm -rf $(OBJS) $(TARGET)
