--- kdebase-workspace-4.0.1/solid/networking/kded/CMakeLists.txt.orig	2008-02-07 21:57:28.000000000 +0100
+++ kdebase-workspace-4.0.1/solid/networking/kded/CMakeLists.txt	2008-02-07 21:59:56.000000000 +0100
@@ -10,7 +10,7 @@
         networkstatus.h NetworkStatusModule)
 
 qt4_add_dbus_adaptor(kded_networkstatus_PART_SRCS
-        ${DBUS_INTERFACES_INSTALL_DIR}/org.kde.Solid.Networking.Client.xml
+        ${DBUS_INTERFACES_SEARCH_DIR}/org.kde.Solid.Networking.Client.xml
         networkstatus.h NetworkStatusModule)
 
 
--- kdebase-workspace-4.0.1/ksmserver/CMakeLists.txt.orig	2008-02-07 22:20:46.000000000 +0100
+++ kdebase-workspace-4.0.1/ksmserver/CMakeLists.txt	2008-02-07 22:21:30.000000000 +0100
@@ -25,7 +25,7 @@
        DEPENDS ${kcminit_adaptor})
 QT4_ADD_DBUS_INTERFACE( ksmserver_KDEINIT_SRCS ${kcminit_xml} kcminit_interface )
 
-set(klauncher_xml  ${DBUS_INTERFACES_INSTALL_DIR}/org.kde.KLauncher.xml)
+set(klauncher_xml  ${DBUS_INTERFACES_SEARCH_DIR}/org.kde.KLauncher.xml)
 QT4_ADD_DBUS_INTERFACE( ksmserver_KDEINIT_SRCS ${klauncher_xml} klauncher_interface )
 
 qt4_add_dbus_adaptor( ksmserver_KDEINIT_SRCS org.kde.KSMServerInterface.xml server.h KSMServer )
