--- efl-1.11.0/configure.ac.orig	2014-08-24 15:17:37.000000000 -0300
+++ efl-1.11.0/configure.ac	2014-08-24 15:17:48.000000000 -0300
@@ -2813,7 +2813,7 @@ EFL_INTERNAL_DEPEND_PKG([ECORE_DRM], [ec
 EFL_INTERNAL_DEPEND_PKG([ECORE_DRM], [eo])
 EFL_INTERNAL_DEPEND_PKG([ECORE_DRM], [eina])
 
-EFL_DEPEND_PKG([ECORE_DRM], [DRM], [libudev >= 148 libdrm >= 2.4 xkbcommon >= 0.3.0 libsystemd-login >= 192 dbus-1])
+EFL_DEPEND_PKG([ECORE_DRM], [DRM], [libudev >= 148 libdrm >= 2.4 xkbcommon >= 0.3.0 dbus-1])
 
 EFL_EVAL_PKGS([ECORE_DRM])
 
