# It is the default Makefile patch for hardcode paths
diff -Nur bwmon-1.3/Makefile bwmon-1.3-fabio.m/Makefile
--- bwmon-1.3/Makefile	2002-05-03 07:12:14.000000000 -0300
+++ bwmon-1.3-fabio.m/Makefile	2008-09-30 01:27:15.000000000 -0300
@@ -8,6 +8,7 @@
 
 RM = /bin/rm
 INSTALL = install
+DESTDIR = /usr/local
 
 MAKE = make
 
@@ -25,6 +26,6 @@
 	fi
 
 install: all
-	$(INSTALL) bwmon /usr/local/bin
+	$(INSTALL) bwmon $(DESTDIR)/bin
 
 
