--- cvs-1.12.13.1/Makefile.am.orig	2017-03-13 14:29:43.853948240 -0300
+++ cvs-1.12.13.1/Makefile.am	2017-03-13 14:30:16.939948861 -0300
@@ -19,8 +19,7 @@
 # This is a legacy variable from b4 Automake
 USOURCE_SUBDIRS = lib $(ZLIB_SUBDIRS) diff src
 # All other subdirs:
-SUBDIRS = $(USOURCE_SUBDIRS) man doc contrib tools \
-	windows-NT os2 emx vms maint-aux
+SUBDIRS = $(USOURCE_SUBDIRS) man doc contrib tools maint-aux
 
 ACLOCAL_AMFLAGS = -I m4
 
--- cvs-1.12.13.1/Makefile.in.orig	2017-03-13 14:29:39.165948152 -0300
+++ cvs-1.12.13.1/Makefile.in	2017-03-13 14:30:25.691949025 -0300
@@ -337,8 +337,7 @@ with_default_rsh = @with_default_rsh@
 # This is a legacy variable from b4 Automake
 USOURCE_SUBDIRS = lib $(ZLIB_SUBDIRS) diff src
 # All other subdirs:
-SUBDIRS = $(USOURCE_SUBDIRS) man doc contrib tools \
-	windows-NT os2 emx vms maint-aux
+SUBDIRS = $(USOURCE_SUBDIRS) man doc contrib tools maint-aux
 
 ACLOCAL_AMFLAGS = -I m4
 EXTRA_DIST = \
