diff -Naur gmp-4.2.4/configure.in gmp-4.2.4-new/configure.in
--- gmp-4.2.4/configure.in	2008-08-01 13:24:04.000000000 -0400
+++ gmp-4.2.4-new/configure.in	2009-03-03 19:19:37.000000000 -0500
@@ -1066,6 +1066,7 @@
 
 
   sh-*-*)   path="sh" ;;
+  sh4-*-*)  path="sh" ;;
   sh2-*-*)  path="sh/sh2 sh" ;;
 
 
diff -Naur gmp-4.2.4/mpn/sh/add_n.s gmp-4.2.4-new/mpn/sh/add_n.s
--- gmp-4.2.4/mpn/sh/add_n.s	2007-08-30 14:31:41.000000000 -0400
+++ gmp-4.2.4-new/mpn/sh/add_n.s	2009-03-03 19:19:37.000000000 -0500
@@ -27,8 +27,8 @@
 
 	.text
 	.align 2
-	.global	___gmpn_add_n
-___gmpn_add_n:
+	.global	__gmpn_add_n
+__gmpn_add_n:
 	mov	#0,r3		! clear cy save reg
 
 Loop:	mov.l	@r5+,r1
diff -Naur gmp-4.2.4/mpn/sh/sub_n.s gmp-4.2.4-new/mpn/sh/sub_n.s
--- gmp-4.2.4/mpn/sh/sub_n.s	2007-08-30 14:31:41.000000000 -0400
+++ gmp-4.2.4-new/mpn/sh/sub_n.s	2009-03-03 19:19:37.000000000 -0500
@@ -27,8 +27,8 @@
 
 	.text
 	.align 2
-	.global	___gmpn_sub_n
-___gmpn_sub_n:
+	.global	__gmpn_sub_n
+__gmpn_sub_n:
 	mov	#0,r3		! clear cy save reg
 
 Loop:	mov.l	@r5+,r1
diff -Naur gmp-4.2.4/mpn/sh/gmp-mparam.h gmp-4.2.4-new/mpn/sh/gmp-mparam.h
--- gmp-4.2.4/mpn/sh/gmp-mparam.h	1969-12-31 19:00:00.000000000 -0500
+++ gmp-4.2.4-new/mpn/sh/gmp-mparam.h	2009-03-03 19:19:37.000000000 -0500
@@ -0,0 +1,23 @@
+/* gmp-mparam.h -- Compiler/machine parameter header file.
+
+Copyright 1991, 1993, 1994, 2001 Free Software Foundation, Inc.
+
+This file is part of the GNU MP Library.
+
+The GNU MP Library is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or (at your
+option) any later version.
+
+The GNU MP Library is distributed in the hope that it will be useful, but
+WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public
+License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the GNU MP Library; see the file COPYING.LIB.  If not, write to
+the Free Software Foundation, Inc., 59 Temple Place - Suite 330, Boston,
+MA 02111-1307, USA. */
+
+#define BITS_PER_MP_LIMB 32
+#define BYTES_PER_MP_LIMB 4
