diff -urNa mozilla-orig/browser/locales/en-US/chrome/branding/brand.dtd mozilla/browser/locales/en-US/chrome/branding/brand.dtd
--- mozilla-orig/browser/locales/en-US/chrome/branding/brand.dtd	2008-06-18 10:06:30.000000000 +1200
+++ mozilla/browser/locales/en-US/chrome/branding/brand.dtd	2008-06-18 10:08:23.000000000 +1200
@@ -1,3 +1,3 @@
-<!ENTITY  brandShortName        "Minefield">
-<!ENTITY  brandFullName         "Minefield">            
-<!ENTITY  vendorShortName       "Mozilla">
+<!ENTITY  brandShortName        "Firefox Community Edition">
+<!ENTITY  brandFullName         "Firefox Community Edition by GoboLinux">
+<!ENTITY  vendorShortName       "GoboLinux">
diff -urNa mozilla-orig/browser/locales/en-US/chrome/branding/brand.properties mozilla/browser/locales/en-US/chrome/branding/brand.properties
--- mozilla-orig/browser/locales/en-US/chrome/branding/brand.properties	2008-06-18 10:06:30.000000000 +1200
+++ mozilla/browser/locales/en-US/chrome/branding/brand.properties	2008-06-18 10:09:04.000000000 +1200
@@ -1,3 +1,3 @@
-brandShortName=Minefield
-brandFullName=Minefield
-vendorShortName=Mozilla
+brandShortName=Firefox Community Edition
+brandFullName=Firefox Community Edition by GoboLinux
+vendorShortName=GoboLinux
