diff -Naurx CVS nucleo/nucleo/plugins/ffmpeg/ffmpegImageSink.cxx nucleo-0.7.2/nucleo/plugins/ffmpeg/ffmpegImageSink.cxx
--- nucleo/nucleo/plugins/ffmpeg/ffmpegImageSink.cxx    2008-06-18 10:25:58.0000 00000 +0000
+++ nucleo-0.7.2/nucleo/plugins/ffmpeg/ffmpegImageSink.cxx 2008-05-23 14:20:05.000000000 +0000
@@ -18,6 +18,8 @@
 #include <stdexcept>
 #include <iomanip>
 
+#include <string.h>
+
 #define NO_SOFTWARE_SCALER 1
 
 // ------------------------------------------------------------------------------------
