--- dpkg-1.18.18/lib/dpkg/Makefile.in.orig	2016-12-24 19:53:58.829767810 -0200
+++ dpkg-1.18.18/lib/dpkg/Makefile.in	2016-12-24 19:54:38.870768561 -0200
@@ -90,8 +90,6 @@ PRE_UNINSTALL = :
 POST_UNINSTALL = :
 build_triplet = @build@
 host_triplet = @host@
-@HAVE_LINKER_VERSION_SCRIPT_TRUE@am__append_1 = \
-@HAVE_LINKER_VERSION_SCRIPT_TRUE@	-Wl,--version-script=$(srcdir)/libdpkg.map
 
 @BUILD_SHARED_TRUE@am__append_2 = \
 @BUILD_SHARED_TRUE@	$(LIBINTL) \
