--- font-bitstream-type1-X11R7.0-1.0.0/Makefile.in	2005-12-21 08:18:16.000000000 +0000
+++ font-bitstream-type1-X11R7.0-1.0.0/Makefile.in.new	2008-06-01 20:54:47.000000000 +0000
@@ -396,7 +396,7 @@
 	$(MKFONTSCALE) $(DESTDIR)$(fontdir)
 	@rm -f $(DESTDIR)$(fontdir)/font.dir
 	$(MKFONTDIR) $(DESTDIR)$(fontdir)
-	$(FCCACHE) $(DESTDIR)$(fontdir)
+#	$(FCCACHE) $(DESTDIR)$(fontdir)
 
 distuninstallcheck:
 	@:
