--- linux-2.6.13.2/Makefile	2005-07-19 22:14:13.000000000 -0300
+++ linux-2.6.13.2-Gobo/Makefile	2005-07-21 15:53:46.000000000 -0300
@@ -1,7 +1,7 @@
 VERSION = 2
 PATCHLEVEL = 6
 SUBLEVEL = 13
-EXTRAVERSION = .2
+EXTRAVERSION = .2-Gobo
 NAME=Woozy Numbat
 
 # *DOCUMENTATION*
@@ -556,7 +556,7 @@
 # makefile but the arguement can be passed to make if needed.
 #
 
-MODLIB	:= $(INSTALL_MOD_PATH)/lib/modules/$(KERNELRELEASE)
+MODLIB	:= /System/Kernel/Modules/$(KERNELRELEASE)
 export MODLIB
 
 
