# fix: compile in kernel >= 2.6.27
diff -ur zaptel-1.2.27-orig/zaptel-base.c zaptel-1.2.27/zaptel-base.c
--- zaptel-1.2.27-orig/zaptel-base.c	2008-08-03 11:52:48.000000000 -0300
+++ zaptel-1.2.27/zaptel-base.c	2010-01-21 12:29:53.000000000 -0200
@@ -156,7 +156,10 @@
 /* udev necessary data structures.  Yeah! */
 #ifdef CONFIG_ZAP_UDEV
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,26)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,27)
+#define CLASS_DEV_CREATE(class, devt, device, name) \
+	device_create(class, device, devt, NULL, "%s", name)
+#elif LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,26)
 #define CLASS_DEV_CREATE(class, devt, device, name) \
 	device_create(class, device, devt, name)
 #elif LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,15)
