--- gtkmathview-0.7.8/mathmlps/main.cc.old	2009-04-27 21:52:54.000000000 +0200
+++ gtkmathview-0.7.8/mathmlps/main.cc	2009-04-27 21:52:41.000000000 +0200
@@ -23,6 +23,7 @@
 #include <config.h>
 
 #include <cassert>
+#include <cstring>
 #include <fstream>
 
 #include <popt.h>
--- gtkmathview-0.7.8/mathmlsvg/SMS.cc.old	2009-04-27 21:54:30.000000000 +0200
+++ gtkmathview-0.7.8/mathmlsvg/SMS.cc	2009-04-27 21:54:21.000000000 +0200
@@ -22,6 +22,7 @@
 
 #include <config.h>
 
+#include <cstring>
 #include <sstream>
 #include "defs.h"
 #include "AbstractLogger.hh"
--- gtkmathview-0.7.8/mathmlsvg/main.cc.old	2009-04-27 21:55:38.000000000 +0200
+++ gtkmathview-0.7.8/mathmlsvg/main.cc	2009-04-27 21:55:31.000000000 +0200
@@ -23,6 +23,7 @@
 #include <config.h>
 
 #include <cassert>
+#include <cstring>
 #include <fstream>
 
 #include <popt.h>
--- gtkmathview-0.7.8/src/backend/ps/T1_FontDataBase.cc.old	2009-04-27 21:56:41.000000000 +0200
+++ gtkmathview-0.7.8/src/backend/ps/T1_FontDataBase.cc	2009-04-27 21:56:32.000000000 +0200
@@ -23,7 +23,7 @@
 #include <t1lib.h>
 #include <config.h>
 #include <map>
-#include <string>
+#include <cstring>
 #include <stdlib.h>
 #include <iostream>
 #include <list>
--- gtkmathview-0.7.8/src/engine/mathml/mathVariantAux.cc.old	2009-04-27 21:57:55.000000000 +0200
+++ gtkmathview-0.7.8/src/engine/mathml/mathVariantAux.cc	2009-04-27 21:57:44.000000000 +0200
@@ -23,6 +23,7 @@
 #include <config.h>
 
 #include <cassert>
+#include <cstring>
 
 #include "mathVariantAux.hh"
 
