Description: Fix mismatching definition of pa_drop_caps

diff -Naur pulseaudio-0.9.10/src/daemon/caps.c pulseaudio-0.9.10-b2/src/daemon/caps.c
--- pulseaudio-0.9.10/src/daemon/caps.c	2008-03-29 21:39:46.000000000 -0300
+++ pulseaudio-0.9.10-b2/src/daemon/caps.c	2008-06-13 18:15:12.000000000 -0300
@@ -131,9 +131,8 @@
     return 0;
 }
 
-int pa_drop_caps(void) {
+void pa_drop_caps(void) {
     pa_drop_root();
-    return 0;
 }
 
 #endif
