--- amarok-2.5.96/CMakeLists.txt.orig	2012-08-05 22:20:30.000000000 -0300
+++ amarok-2.5.96/CMakeLists.txt	2012-08-05 22:21:10.000000000 -0300
@@ -29,13 +29,13 @@ if(TAGLIB_FOUND)
     set(CMAKE_REQUIRED_INCLUDES "${TAGLIB_INCLUDES}")
     set(CMAKE_REQUIRED_LIBRARIES "${TAGLIB_LIBRARIES}")
 
-    check_cxx_source_compiles("#include <asftag.h>
+    check_cxx_source_compiles("#include <taglib/asftag.h>
     int main() { TagLib::ASF::Tag tag; return 0;}" TAGLIB_ASF_FOUND)
     if( NOT TAGLIB_ASF_FOUND )
         message(FATAL_ERROR "TagLib does not have ASF support compiled in.")
     endif( NOT TAGLIB_ASF_FOUND )
 
-    check_cxx_source_compiles("#include <mp4tag.h>
+    check_cxx_source_compiles("#include <taglib/mp4tag.h>
     int main() { TagLib::MP4::Tag tag(0, 0); return 0;}" TAGLIB_MP4_FOUND)
     if( NOT TAGLIB_MP4_FOUND )
         message(FATAL_ERROR "TagLib does not have MP4 support compiled in.")
