diff -Nur kdebase-3.2.3/kcheckpass/Makefile.am kdebase-3.2.3-lucasvr/kcheckpass/Makefile.am
--- kdebase-3.2.3/kcheckpass/Makefile.am	2003-11-30 07:42:21.000000000 -0200
+++ kdebase-3.2.3-lucasvr/kcheckpass/Makefile.am	2004-08-29 21:49:18.000000000 -0300
@@ -21,5 +21,5 @@
 	-@test -n "$(DESTDIR)" || test -z "$(PAM)" || $(top_srcdir)/mkpamserv $(PAM)
 
 install-exec-hook:
-	@(chown root $(DESTDIR)$(bindir)/kcheckpass && chmod 4755 $(DESTDIR)$(bindir)/kcheckpass) \
+	@(chown 0 $(DESTDIR)$(bindir)/kcheckpass && chmod 4755 $(DESTDIR)$(bindir)/kcheckpass) \
 	|| echo "Error: Could not install kcheckpass as setuid root (possibly you won't be able to unlock)!!\n"
diff -Nur kdebase-3.2.3/kcheckpass/Makefile.in kdebase-3.2.3-lucasvr/kcheckpass/Makefile.in
--- kdebase-3.2.3/kcheckpass/Makefile.in	2004-06-02 04:42:06.000000000 -0300
+++ kdebase-3.2.3-lucasvr/kcheckpass/Makefile.in	2004-08-29 21:49:51.000000000 -0300
@@ -835,7 +835,7 @@
 	-@test -n "$(DESTDIR)" || test -z "$(PAM)" || $(top_srcdir)/mkpamserv $(PAM)
 
 install-exec-hook:
-	@(chown root $(DESTDIR)$(bindir)/kcheckpass && chmod 4755 $(DESTDIR)$(bindir)/kcheckpass) \
+	@(chown 0 $(DESTDIR)$(bindir)/kcheckpass && chmod 4755 $(DESTDIR)$(bindir)/kcheckpass) \
 	|| echo "Error: Could not install kcheckpass as setuid root (possibly you won't be able to unlock)!!\n"
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
