Index: kdebase/libkonq/knewmenu.cc
===================================================================
--- kdebase/libkonq/knewmenu.cc
+++ kdebase/libkonq/knewmenu.cc
@@ -19,6 +19,8 @@
 
 #include <qdir.h>
 
+#include <unistd.h>
+
 #include <kdebug.h>
 #include <kdesktopfile.h>
 #include <kdirwatch.h>
@@ -465,6 +467,19 @@ void KNewMenu::slotNewFile()
 
         KURL uSrc;
         uSrc.setPath( src );
+        if ( uSrc.isLocalFile() )
+        {
+            // In case the templates/.sources directory contains symlinks, resolve
+            // them to the target files. Fixes KDE bug #149628.
+            // Uses code from KFileItem::linkDest modified to suit by Michael Homer.
+            char buf[1000];
+            int n = readlink( QFile::encodeName( src ), buf, sizeof(buf)-1 );
+            if ( n != -1 )
+            {
+                buf[ n ] = 0;
+                uSrc.setPath( QFile::decodeName( buf ) );
+            }
+        }
         //kdDebug(1203) << "KNewMenu : KIO::copyAs( " << uSrc.url() << ", " << dest.url() << ")" << endl;
         KIO::CopyJob * job = KIO::copyAs( uSrc, dest );
         job->setDefaultPermissions( true );
