--- rpm-5.3.5/rpmdb/rpmdb.h.orig	2016-11-21 17:28:15.132279348 -0200
+++ rpm-5.3.5/rpmdb/rpmdb.h	2016-11-21 17:31:18.838282794 -0200
@@ -15,7 +15,7 @@
 
 #if defined(_RPMDB_INTERNAL)
 #if defined(WITH_DB)
-#include <db51/db.h>
+#include <db.h>
 #else
 #include "db_emu.h"
 #endif
--- rpm-5.3.5/configure.ac.orig	2016-11-21 18:15:24.260332424 -0200
+++ rpm-5.3.5/configure.ac	2016-11-21 18:22:22.354340268 -0200
@@ -561,7 +561,7 @@ dnl # platform specific checks
 AC_SEARCH_LIBS([strerror],[cposix])
 
 dnl # standard checks
-AM_C_PROTOTYPES
+dnl AM_C_PROTOTYPES
 AC_C_INLINE
 AC_C_BIGENDIAN
 
@@ -1373,7 +1373,7 @@ dnl # Berkeley-DB & SQLite
 DBLIBSRCS=""
 RPM_CHECK_LIB(
     [Berkeley-DB], [db],
-    [db-5.1], [db_create], [db51/db.h],
+    [db], [db_create], [db.h],
     [yes,external], [db3],
     [ DBLIBSRCS="$DBLIBSRCS db3.c"
       AM_CONDITIONAL(WITH_DB, [ true ])
