diff -Naur netkit-telnet-0.17/telnet/main.cc netkit-telnet-0.17.new/telnet/main.cc
--- netkit-telnet-0.17/telnet/main.cc	1999-08-01 07:06:37.000000000 +0200
+++ netkit-telnet-0.17.new/telnet/main.cc	2008-02-18 20:40:27.000000000 +0100
@@ -46,6 +46,7 @@
 #include <sys/types.h>
 #include <getopt.h>
 #include <string.h>
+#include <stdlib.h>
 
 #include "ring.h"
 #include "externs.h"
diff -Naur netkit-telnet-0.17/telnet/netlink.cc netkit-telnet-0.17.new/telnet/netlink.cc
--- netkit-telnet-0.17/telnet/netlink.cc	2000-07-23 06:16:25.000000000 +0200
+++ netkit-telnet-0.17.new/telnet/netlink.cc	2008-02-18 20:55:13.000000000 +0100
@@ -8,6 +8,7 @@
 #include <sys/ioctl.h>
 #include <sys/time.h>
 #include <netdb.h>
+#include <string.h>
 #include "netlink.h"
 #include "proto.h"
 #include "ring.h"
diff -Naur netkit-telnet-0.17/telnet/network.cc netkit-telnet-0.17.new/telnet/network.cc
--- netkit-telnet-0.17/telnet/network.cc	1996-08-13 10:09:58.000000000 +0200
+++ netkit-telnet-0.17.new/telnet/network.cc	2008-02-18 20:42:04.000000000 +0100
@@ -42,6 +42,7 @@
 #include <sys/time.h>
 #include <errno.h>
 #include <arpa/telnet.h>
+#include <stdlib.h>
 
 #include "ring.h"
 #include "defines.h"
diff -Naur netkit-telnet-0.17/telnet/terminal.cc netkit-telnet-0.17.new/telnet/terminal.cc
--- netkit-telnet-0.17/telnet/terminal.cc	1999-12-12 20:48:05.000000000 +0100
+++ netkit-telnet-0.17.new/telnet/terminal.cc	2008-02-18 20:50:03.000000000 +0100
@@ -45,6 +45,8 @@
 #include <signal.h>
 #include <errno.h>
 #include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
 
 #include "ring.h"
 #include "defines.h"
diff -Naur netkit-telnet-0.17/telnet/utilities.cc netkit-telnet-0.17.new/telnet/utilities.cc
--- netkit-telnet-0.17/telnet/utilities.cc	1999-12-12 16:33:40.000000000 +0100
+++ netkit-telnet-0.17.new/telnet/utilities.cc	2008-02-18 20:51:33.000000000 +0100
@@ -47,6 +47,8 @@
 #include <sys/socket.h>
 #include <unistd.h>
 #include <ctype.h>
+#include <stdlib.h>
+#include <string.h>
 
 #include "ring.h"
 #include "defines.h"
