--- font-misc-ethiopic-X11R7.0-1.0.0/Makefile.in	2005-12-21 08:19:34.000000000 +0000
+++ font-misc-ethiopic-X11R7.0-1.0.0/Makefile.in.new	2008-06-01 21:14:10.000000000 +0000
@@ -400,12 +400,12 @@
 	$(MKFONTSCALE) $(DESTDIR)$(ttffontdir)
 	@rm -f $(DESTDIR)$(ttffontdir)/font.dir
 	$(MKFONTDIR) $(DESTDIR)$(ttffontdir)
-	$(FCCACHE) $(DESTDIR)$(ttffontdir)
+#	$(FCCACHE) $(DESTDIR)$(ttffontdir)
 	@rm -f $(DESTDIR)$(otffontdir)/fonts.scale
 	$(MKFONTSCALE) $(DESTDIR)$(otffontdir)
 	@rm -f $(DESTDIR)$(otffontdir)/font.dir
 	$(MKFONTDIR) $(DESTDIR)$(otffontdir)
-	$(FCCACHE) $(DESTDIR)$(otffontdir)
+#	$(FCCACHE) $(DESTDIR)$(otffontdir)
 
 distuninstallcheck:
 	@:
