
--- gpm-1.20.3/configure.in.old	2008-05-23 13:37:30.000000000 +0000
+++ gpm-1.20.3/configure.in	2008-05-23 13:35:25.000000000 +0000
@@ -61,6 +61,17 @@
 
 AC_CHECK_HEADERS(syslog.h linux/input.h linux/joystick.h ncurses.h ncurses/curses.h curses.h)
 
+# Check for ucred
+AC_MSG_CHECKING(for struct ucred)
+AC_TRY_LINK([
+#include <sys/ucred.h>
+#include <sys/socket.h>
+],[
+struct ucred sucred; sucred.pid=0;
+],[has_ucred=yes],[has_ucred=no])
+AC_MSG_RESULT($has_ucred)
+if test ${has_ucred} = no ; then CFLAGS="$CFLAGS -DNEED_UCRED"; fi
+
 AC_ARG_WITH(curses,
 [  --without-curses        disable curses support even if curses found])
 
--- gpm-1.20.3/src/daemon/processconn.c	2008-05-23 13:22:20.000000000 +0000
+++ gpm-1.20.3/src/daemon/processconn.c.new	2008-05-23 13:41:26.000000000 +0000
@@ -101,6 +101,14 @@
    uid = statbuf.st_uid;      /* owner of socket */
 #else
    {
+#ifdef NEED_UCRED /* from autoconf */
+struct ucred
+   {
+      pid_t pid; /* PID of sending process. */
+      uid_t uid; /* UID of sending process. */
+      gid_t gid; /* GID of sending process. */
+   };
+#endif /* NEED_UCRED */
       struct ucred sucred;
       socklen_t credlen = sizeof(struct ucred);
 
