--- linux-2.4.26/fs/devfs/base.c.orig	2004-04-25 20:43:29.000000000 -0300
+++ linux-2.4.26/fs/devfs/base.c	2004-04-25 20:43:53.000000000 -0300
@@ -3549,8 +3549,8 @@
     int err;
 
     if ( !(boot_options & OPTION_MOUNT) ) return;
-    err = do_mount ("none", "/dev", "devfs", 0, NULL);
-    if (err == 0) printk (KERN_INFO "Mounted devfs on /dev\n");
+    err = do_mount ("none", "/System/Kernel/Devices", "devfs", 0, NULL);
+    if (err == 0) printk (KERN_INFO "Mounted devfs on /System/Kernel/Devices\n");
     else PRINTK ("(): unable to mount devfs, err: %d\n", err);
 }   /*  End Function mount_devfs_fs  */
 
