Do not start the mdns daemon during a build.

diff -Nur mDNSResponder-878.1.1.old/mDNSPosix/Makefile mDNSResponder-878.1.1/mDNSPosix/Makefile
--- mDNSResponder-878.1.1.old/mDNSPosix/Makefile	2016-04-24 12:07:24.000000000 -0300
+++ mDNSResponder-878.1.1/mDNSPosix/Makefile	2017-11-26 19:32:43.392186202 -0200
@@ -299,9 +299,7 @@
 
 # Note: If daemon already installed, we make sure it's stopped before overwriting it
 $(INSTBASE)/sbin/mdnsd: $(BUILDDIR)/mdnsd $(STARTUPSCRIPTDIR)/$(STARTUPSCRIPTNAME)
-	@if test -x $@; then $(STARTUPSCRIPTDIR)/$(STARTUPSCRIPTNAME) stop; fi
 	$(CP) $< $@
-	@$(STARTUPSCRIPTDIR)/$(STARTUPSCRIPTNAME) start
 
 $(INSTBASE)/lib/libdns_sd.$(LDSUFFIX).$(LIBVERS): $(BUILDDIR)/libdns_sd.$(LDSUFFIX)
 	$(CP) $< $@
