diff -Naur mozilla/browser/locales/en-US/chrome/global/brand.dtd mozilla.new/browser/locales/en-US/chrome/global/brand.dtd
--- mozilla/browser/locales/en-US/chrome/global/brand.dtd	2004-10-16 02:04:14.000000000 +0200
+++ mozilla.new/browser/locales/en-US/chrome/global/brand.dtd	2005-09-17 10:52:42.000000000 +0200
@@ -1,5 +1,5 @@
-<!ENTITY  brandShortName        "Firefox">
-<!ENTITY  brandFullName         "Mozilla Firefox">            
+<!ENTITY  brandShortName        "Firefox Community Edition by GoboLinux">
+<!ENTITY  brandFullName         "Mozilla Firefox Community Edition by GoboLinux">
 <!ENTITY  vendorShortName       "Mozilla">
 
 <!ENTITY  releaseURL            "http://www.mozilla.org/products/firefox/releases/1.0.html">
diff -Naur mozilla/browser/locales/en-US/chrome/global/brand.properties mozilla.new/browser/locales/en-US/chrome/global/brand.properties
--- mozilla/browser/locales/en-US/chrome/global/brand.properties	2004-07-16 21:57:59.000000000 +0200
+++ mozilla.new/browser/locales/en-US/chrome/global/brand.properties	2005-09-17 10:53:06.000000000 +0200
@@ -1,3 +1,3 @@
-brandShortName=Firefox
-brandFullName=Mozilla Firefox
+brandShortName=Firefox Community Edition by GoboLinux
+brandFullName=Mozilla Firefox Community Edition by GoboLinux
 vendorShortName=Mozilla
