--- Mesa-7.0.2/configs/linux-dri-cell.orig	1970-01-01 00:00:00.000000000 +0000
+++ Mesa-7.0.2/configs/linux-dri-cell	2008-02-15 18:53:31.000000000 +0000
@@ -0,0 +1,17 @@
+# -*-makefile-*-
+# Configuration for linux-dri: Linux DRI hardware drivers for XFree86 & others
+
+include $(TOP)/configs/linux-dri
+
+CONFIG_NAME = linux-dri-cell
+
+OPT_FLAGS = -O3 -mcpu=cell -mabi=altivec
+PIC_FLAGS = -fPIC
+
+ASM_FLAGS = -DUSE_PPC_ASM -DUSE_VMX_ASM
+ASM_SOURCES = $(PPC_SOURCES)
+
+# Build only the drivers for cards that exist on PowerPC.  At some point MGA
+# will be added, but not yet.
+DRI_DIRS = mach64 r128 r200 r300 radeon tdfx
+
