Fix for missing MOD_NANO
From: https://support.ntp.org/bugs/show_bug.cgi?id=1219
diff -aur ntp-stable/ntpd/ntp_loopfilter.c ntp-stable+ch/ntpd/ntp_loopfilter.c
--- ntp-stable/ntpd/ntp_loopfilter.c	2007-02-24 02:07:04.000000000 -0800
+++ ntp-stable+ch/ntpd/ntp_loopfilter.c	2009-11-09 13:21:52.000000000 -0800
@@ -177,6 +177,9 @@
 static struct sigaction newsigsys; /* new sigaction status */
 static sigjmp_buf env;		/* environment var. for pll_trap() */
 #endif /* SIGSYS */
+#if defined(STA_NANO) && !defined(MOD_NANO)
+#define MOD_NANO ADJ_NANO	/* current linux headers */
+#endif
 #endif /* KERNEL_PLL */
 
 /*
