--- neon-0.29.1/neon.pc.in	2006-06-26 12:01:29.000000000 +0200
+++ neon-0.29.1/neon.pc.in.new	2009-12-16 20:42:01.000000000 +0100
@@ -1,11 +1,11 @@
 prefix=@prefix@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
-includedir=@includedir@
+includedir=@includedir@/neon
 
 Name: neon
 Description: HTTP/WebDAV client library
 Version: @NEON_VERSION@
 Libs.private: @NEON_LIBS@
 Libs: -L${libdir} -lneon @NEON_PC_LIBS@
-Cflags: -I${includedir}/neon @NEON_CFLAGS@
+Cflags: -I${includedir} @NEON_CFLAGS@
