--- nethack-3.4.3-orig/include/unixconf.h	2003-12-08 00:39:13.000000000 +0100
+++ nethack-3.4.3/include/unixconf.h	2008-03-15 05:39:56.000000000 +0100
@@ -37,7 +37,9 @@
 #define NETWORK		/* if running on a networked system */
 			/* e.g. Suns sharing a playground through NFS */
 /* #define SUNOS4 */	/* SunOS 4.x */
-/* #define LINUX */	/* Another Unix clone */
+#define LINUX 	/* Another Unix clone */
+#define SHOW_BORN
+#define SORTLOOT
 /* #define CYGWIN32 */	/* Unix on Win32 -- use with case sensitive defines */
 /* #define GENIX */	/* Yet Another Unix Clone */
 /* #define HISX */	/* Bull Unix for XPS Machines */
@@ -132,7 +132,7 @@
  * "extra output" method is used, but not all systems provide access to
  * a fine-grained timer.
  */
-/* #define TIMED_DELAY */	/* usleep() */
+#define TIMED_DELAY 	/* usleep() */
 #endif
 
 /*
