diff -Naurp mozilla-orig/toolkit/xre/nsAppRunner.cpp mozilla/toolkit/xre/nsAppRunner.cpp
--- mozilla-orig/toolkit/xre/nsAppRunner.cpp	2008-11-17 08:36:43.000000000 -0600
+++ mozilla/toolkit/xre/nsAppRunner.cpp	2009-03-01 20:35:11.000000000 -0600
@@ -1459,7 +1459,7 @@ XRE_GetBinaryPath(const char* argv0, nsI
   if (NS_FAILED(rv))
     return rv;
 
-#elif
+#else
 #error Oops, you need platform-specific code here
 #endif
 
