--- libcaca-0.9/doc/Makefile.am.orig	2008-04-12 16:26:25.000000000 +0200
+++ libcaca-0.9/doc/Makefile.am	2008-04-12 16:26:44.000000000 +0200
@@ -45,7 +45,7 @@
 	cp `find html -name '*.html' -o -name '*.gif' -o -name '*.png'` \
 	  $(DESTDIR)$(datadir)/doc/libcaca-dev/html
 	$(mkinstalldirs) $(DESTDIR)$(mandir)/man3
-	cp man/man3/*.3caca $(DESTDIR)$(mandir)/man3
+	cp man/man3caca/*.3caca $(DESTDIR)$(mandir)/man3
 	$(mkinstalldirs) $(DESTDIR)$(mandir)/man1
 	rm -f $(DESTDIR)$(mandir)/man1/cacaball.1
 	ln -s cacademo.1 $(DESTDIR)$(mandir)/man1/cacaball.1
--- libcaca-0.9/doc/Makefile.in.orig	2008-04-12 16:32:39.000000000 +0200
+++ libcaca-0.9/doc/Makefile.in	2008-04-12 16:32:54.000000000 +0200
@@ -405,7 +405,7 @@
 @BUILD_DOCUMENTATION_TRUE@	cp `find html -name '*.html' -o -name '*.gif' -o -name '*.png'` \
 @BUILD_DOCUMENTATION_TRUE@	  $(DESTDIR)$(datadir)/doc/libcaca-dev/html
 @BUILD_DOCUMENTATION_TRUE@	$(mkinstalldirs) $(DESTDIR)$(mandir)/man3
-@BUILD_DOCUMENTATION_TRUE@	cp man/man3/*.3caca $(DESTDIR)$(mandir)/man3
+@BUILD_DOCUMENTATION_TRUE@	cp man/man3caca/*.3caca $(DESTDIR)$(mandir)/man3
 @BUILD_DOCUMENTATION_TRUE@	$(mkinstalldirs) $(DESTDIR)$(mandir)/man1
 @BUILD_DOCUMENTATION_TRUE@	rm -f $(DESTDIR)$(mandir)/man1/cacaball.1
 @BUILD_DOCUMENTATION_TRUE@	ln -s cacademo.1 $(DESTDIR)$(mandir)/man1/cacaball.1
