Index: README
===================================================================
RCS file: /home/mleisher/cvsrep/otf2bdf/README,v
retrieving revision 1.3
retrieving revision 1.4
diff -u -r1.3 -r1.4
--- README	7 Nov 2005 23:34:51 -0000	1.3
+++ README	9 Nov 2005 17:27:52 -0000	1.4
@@ -1,5 +1,5 @@
 #
-# $Id: README,v 1.3 2005/11/07 23:34:51 mleisher Exp $
+# $Id: README,v 1.4 2005/11/09 17:27:52 mleisher Exp $
 #
 
 #
@@ -127,6 +127,9 @@
 
   Christos Tountas <cvt@sprynet.com> for finding a buffer overflow problem.
 
+  Nelson Beebe <beebe@math.utah.edu> for finding problems with the FreeType
+  includes, the configure setup, and a compilation warning.
+
 CHANGES
 -------
 Version 3.0
Index: Makefile.in
===================================================================
RCS file: /home/mleisher/cvsrep/otf2bdf/Makefile.in,v
retrieving revision 1.1
retrieving revision 1.2
diff -u -r1.1 -r1.2
--- Makefile.in	21 Sep 2005 21:40:39 -0000	1.1
+++ Makefile.in	9 Nov 2005 17:29:06 -0000	1.2
@@ -1,5 +1,5 @@
 #
-# $Id: Makefile.in,v 1.1.1.1 2005/09/21 21:40:39 mleisher Exp $
+# $Id: Makefile.in,v 1.2 2005/11/09 17:29:06 mleisher Exp $
 #
 
 #
@@ -39,6 +39,7 @@
 #
 INCS = @CPPFLAGS@
 LIBS = @LIBS@
+LDFLAGS = @LDFLAGS@
 
 prefix = @prefix@
 exec_prefix = @exec_prefix@
@@ -48,7 +49,7 @@
 all: otf2bdf
 
 otf2bdf: $(OBJS)
-	$(CC) $(STATIC) $(CFLAGS) -o otf2bdf $(OBJS) $(LIBS)
+	$(CC) $(STATIC) $(LDFLAGS) -o otf2bdf $(OBJS) $(LIBS)
 
 clean:
 	$(RM) -f *.o *BAK *CKP *~ a.out core
Index: otf2bdf.c
===================================================================
RCS file: /home/mleisher/cvsrep/otf2bdf/otf2bdf.c,v
retrieving revision 1.3
retrieving revision 1.4
diff -u -r1.3 -r1.4
--- otf2bdf.c	7 Nov 2005 23:35:24 -0000	1.3
+++ otf2bdf.c	9 Nov 2005 17:29:44 -0000	1.4
@@ -21,9 +21,9 @@
  */
 #ifndef lint
 #ifdef __GNUC__
-static char rcsid[] __attribute__ ((unused)) = "$Id: otf2bdf.c,v 1.3 2005/11/07 23:35:24 mleisher Exp $";
+static char rcsid[] __attribute__ ((unused)) = "$Id: otf2bdf.c,v 1.4 2005/11/09 17:29:44 mleisher Exp $";
 #else
-static char rcsid[] = "$Id: otf2bdf.c,v 1.3 2005/11/07 23:35:24 mleisher Exp $";
+static char rcsid[] = "$Id: otf2bdf.c,v 1.4 2005/11/09 17:29:44 mleisher Exp $";
 #endif
 #endif
 
@@ -38,7 +38,8 @@
 
 #include <string.h>
 
-#include <freetype/freetype.h>
+#include <ft2build.h>
+#include FT_FREETYPE_H
 #include FT_GLYPH_H
 #include FT_SFNT_NAMES_H
 #include FT_TRUETYPE_TABLES_H
@@ -520,9 +521,13 @@
         for (j = 0; j < 63 && *np != '-'; np++, j++)
           encoding[j] = *np;
         encoding[j] = 0;
+        /*
+         * Typecast the result of 14-strlen(). This returns a size_t on
+         * some platforms and causes a compilation warning.
+         */
         printf("%s%.*s%s%.*s(-pid %hd -eid %hd)\n",
-               platform, 14 - strlen(platform), spaces,
-               encoding, 14 - strlen(encoding), spaces, pid, eid);
+               platform, (int) (14 - strlen(platform)), spaces,
+               encoding, (int) (14 - strlen(encoding)), spaces, pid, eid);
     }
 }
 
Index: otf2bdf.man
===================================================================
RCS file: /home/mleisher/cvsrep/otf2bdf/otf2bdf.man,v
retrieving revision 1.3
retrieving revision 1.4
diff -u -r1.3 -r1.4
--- otf2bdf.man	7 Nov 2005 23:34:51 -0000	1.3
+++ otf2bdf.man	9 Nov 2005 17:30:26 -0000	1.4
@@ -1,5 +1,5 @@
 .\"
-.\" $Id: otf2bdf.man,v 1.3 2005/11/07 23:34:51 mleisher Exp $
+.\" $Id: otf2bdf.man,v 1.4 2005/11/09 17:30:26 mleisher Exp $
 .\"
 .TH OTF2BDF 1 "07 November 2005" "Freetype Version 2"
 .SH NAME 
@@ -204,6 +204,8 @@
 xmbdfed.
 .br
 Christos Tountas <cvt@sprynet.com> for finding problems.
+.br
+Nelson Beebe <beebe@math.utah.edu> for finding problems.
 .SH AUTHOR
 Mark Leisher
 .br
