diff -Naur pilot-link-0.12.2/doc/60-libpisock.rules pilot-link-0.12.2-new/doc/60-libpisock.rules
--- pilot-link-0.12.2/doc/60-libpisock.rules	2007-02-09 12:08:06.000000000 -0200
+++ pilot-link-0.12.2-new/doc/60-libpisock.rules	2007-09-02 18:34:25.000000000 -0300
@@ -3,43 +3,43 @@
 SUBSYSTEMS!="usb", ACTION!="add", GOTO="libpisock_rules_end"
 
 # Sony handheld devices
-ATTRS{idVendor}=="054c", ATTRS{idProduct}=="0038", GROUP="dialout", MODE="0664"
-ATTRS{idVendor}=="054c", ATTRS{idProduct}=="0066", GROUP="dialout", MODE="0664"
-ATTRS{idVendor}=="054c", ATTRS{idProduct}=="0095", GROUP="dialout", MODE="0664"
-ATTRS{idVendor}=="054c", ATTRS{idProduct}=="009a", GROUP="dialout", MODE="0664"
-ATTRS{idVendor}=="054c", ATTRS{idProduct}=="00da", GROUP="dialout", MODE="0664"
-ATTRS{idVendor}=="054c", ATTRS{idProduct}=="00e9", GROUP="dialout", MODE="0664"
-ATTRS{idVendor}=="054c", ATTRS{idProduct}=="0144", GROUP="dialout", MODE="0664"
-ATTRS{idVendor}=="054c", ATTRS{idProduct}=="0169", GROUP="dialout", MODE="0664"
+ATTRS{idVendor}=="054c", ATTRS{idProduct}=="0038", GROUP="console", MODE="0664"
+ATTRS{idVendor}=="054c", ATTRS{idProduct}=="0066", GROUP="console", MODE="0664"
+ATTRS{idVendor}=="054c", ATTRS{idProduct}=="0095", GROUP="console", MODE="0664"
+ATTRS{idVendor}=="054c", ATTRS{idProduct}=="009a", GROUP="console", MODE="0664"
+ATTRS{idVendor}=="054c", ATTRS{idProduct}=="00da", GROUP="console", MODE="0664"
+ATTRS{idVendor}=="054c", ATTRS{idProduct}=="00e9", GROUP="console", MODE="0664"
+ATTRS{idVendor}=="054c", ATTRS{idProduct}=="0144", GROUP="console", MODE="0664"
+ATTRS{idVendor}=="054c", ATTRS{idProduct}=="0169", GROUP="console", MODE="0664"
 
 # Handspring handheld devices
-ATTRS{idVendor}=="082d", ATTRS{idProduct}=="0100", GROUP="dialout", MODE="0664"
-ATTRS{idVendor}=="082d", ATTRS{idProduct}=="0200", GROUP="dialout", MODE="0664"
-ATTRS{idVendor}=="082d", ATTRS{idProduct}=="0300", GROUP="dialout", MODE="0664"
+ATTRS{idVendor}=="082d", ATTRS{idProduct}=="0100", GROUP="console", MODE="0664"
+ATTRS{idVendor}=="082d", ATTRS{idProduct}=="0200", GROUP="console", MODE="0664"
+ATTRS{idVendor}=="082d", ATTRS{idProduct}=="0300", GROUP="console", MODE="0664"
 
 # Palm, Inc. and palmOne handheld devices
-ATTRS{idVendor}=="04e8", ATTRS{idProduct}=="6601", GROUP="dialout", MODE="0664"
-ATTRS{idVendor}=="04e8", ATTRS{idProduct}=="8001", GROUP="dialout", MODE="0664"
-ATTRS{idVendor}=="0830", ATTRS{idProduct}=="0001", GROUP="dialout", MODE="0664"
-ATTRS{idVendor}=="0830", ATTRS{idProduct}=="0002", GROUP="dialout", MODE="0664"
-ATTRS{idVendor}=="0830", ATTRS{idProduct}=="0003", GROUP="dialout", MODE="0664"
-ATTRS{idVendor}=="0830", ATTRS{idProduct}=="0020", GROUP="dialout", MODE="0664"
-ATTRS{idVendor}=="0830", ATTRS{idProduct}=="0031", GROUP="dialout", MODE="0664"
-ATTRS{idVendor}=="0830", ATTRS{idProduct}=="0040", GROUP="dialout", MODE="0664"
-ATTRS{idVendor}=="0830", ATTRS{idProduct}=="0050", GROUP="dialout", MODE="0664"
-ATTRS{idVendor}=="0830", ATTRS{idProduct}=="0060", GROUP="dialout", MODE="0664"
-ATTRS{idVendor}=="0830", ATTRS{idProduct}=="0061", GROUP="dialout", MODE="0664"
-ATTRS{idVendor}=="0830", ATTRS{idProduct}=="0070", GROUP="dialout", MODE="0664"
-ATTRS{idVendor}=="0830", ATTRS{idProduct}=="0080", GROUP="dialout", MODE="0664"
+ATTRS{idVendor}=="04e8", ATTRS{idProduct}=="6601", GROUP="console", MODE="0664"
+ATTRS{idVendor}=="04e8", ATTRS{idProduct}=="8001", GROUP="console", MODE="0664"
+ATTRS{idVendor}=="0830", ATTRS{idProduct}=="0001", GROUP="console", MODE="0664"
+ATTRS{idVendor}=="0830", ATTRS{idProduct}=="0002", GROUP="console", MODE="0664"
+ATTRS{idVendor}=="0830", ATTRS{idProduct}=="0003", GROUP="console", MODE="0664"
+ATTRS{idVendor}=="0830", ATTRS{idProduct}=="0020", GROUP="console", MODE="0664"
+ATTRS{idVendor}=="0830", ATTRS{idProduct}=="0031", GROUP="console", MODE="0664"
+ATTRS{idVendor}=="0830", ATTRS{idProduct}=="0040", GROUP="console", MODE="0664"
+ATTRS{idVendor}=="0830", ATTRS{idProduct}=="0050", GROUP="console", MODE="0664"
+ATTRS{idVendor}=="0830", ATTRS{idProduct}=="0060", GROUP="console", MODE="0664"
+ATTRS{idVendor}=="0830", ATTRS{idProduct}=="0061", GROUP="console", MODE="0664"
+ATTRS{idVendor}=="0830", ATTRS{idProduct}=="0070", GROUP="console", MODE="0664"
+ATTRS{idVendor}=="0830", ATTRS{idProduct}=="0080", GROUP="console", MODE="0664"
 
 # Garmin iQue 3600
-ATTRS{idVendor}=="091e", ATTRS{idProduct}=="0004", GROUP="dialout", MODE="0664"
-ATTRS{idVendor}=="0c88", ATTRS{idProduct}=="0021", GROUP="dialout", MODE="0664"
+ATTRS{idVendor}=="091e", ATTRS{idProduct}=="0004", GROUP="console", MODE="0664"
+ATTRS{idVendor}=="0c88", ATTRS{idProduct}=="0021", GROUP="console", MODE="0664"
 
 # Tapwave Zodiac 1 & 2
-ATTRS{idVendor}=="12ef", ATTRS{idProduct}=="0100", GROUP="dialout", MODE="0664"
+ATTRS{idVendor}=="12ef", ATTRS{idProduct}=="0100", GROUP="console", MODE="0664"
 
 # Unknown device (can someone confirm?)
-ATTRS{idVendor}=="4766", ATTRS{idProduct}=="0001", GROUP="dialout", MODE="0664"
+ATTRS{idVendor}=="4766", ATTRS{idProduct}=="0001", GROUP="console", MODE="0664"
 
 LABEL="libpisock_rules_end"
