--- pinentry-0.7.5/gtk+-2/gtksecentry.c.old	2008-07-13 11:32:38.000000000 +0000
+++ pinentry-0.7.5/gtk+-2/gtksecentry.c	2008-07-13 11:32:19.000000000 +0000
@@ -270,7 +270,7 @@
 
 
 gpointer
-g_malloc(gulong size)
+g_malloc(gsize size)
 {
     gpointer p;
 
@@ -288,7 +288,7 @@
 }
 
 gpointer
-g_malloc0(gulong size)
+g_malloc0(gsize size)
 {
     gpointer p;
 
@@ -308,7 +308,7 @@
 }
 
 gpointer
-g_realloc(gpointer mem, gulong size)
+g_realloc(gpointer mem, gsize size)
 {
     gpointer p;
 
