--- linux-2.6.10-rc3/Makefile.orig	2004-12-04 18:55:41.000000000 -0200
+++ linux-2.6.10-rc3/Makefile	2004-12-04 18:56:09.000000000 -0200
@@ -1,7 +1,7 @@
 VERSION = 2
 PATCHLEVEL = 6
 SUBLEVEL = 10
-EXTRAVERSION =
+EXTRAVERSION = -Gobo
 NAME=Woozy Numbat
 
 # *DOCUMENTATION*
@@ -551,7 +551,7 @@
 # makefile but the arguement can be passed to make if needed.
 #
 
-MODLIB	:= $(INSTALL_MOD_PATH)/lib/modules/$(KERNELRELEASE)
+MODLIB	:= /System/Kernel/Modules/$(KERNELRELEASE)
 export MODLIB
 
 
