--- sidplay-libs-2.1.1.orig/libsidutils/include/sidplay/utils/SidUsage.h	2006-03-19 04:19:17.000000000 +0000
+++ sidplay-libs-2.1.1/libsidutils/include/sidplay/utils/SidUsage.h	2006-03-19 04:19:28.000000000 +0000
@@ -33,7 +33,7 @@
     uint_least16_t length;  // usage scan length
 
     // Copy common parts of basic usage to extended usage.
-    sid2_usage_t &sid2_usage_t::operator= (const sid_usage_t &usage)
+    sid2_usage_t &operator= (const sid_usage_t &usage)
     {
         *((sid_usage_t *) this) = usage;
         return *this;
