--- openldap-2.3.39/configure.in	2007-06-10 18:39:53.000000000 +0000
+++ openldap-2.3.39/configure.in.new	2008-05-23 20:45:49.000000000 +0000
@@ -930,6 +930,17 @@
 #endif
 ])
 
+# Check for ucred
+AC_MSG_CHECKING(for struct ucred)
+AC_TRY_LINK([
+#include <sys/ucred.h>
+#include <sys/socket.h>
+],[
+struct ucred sucred; sucred.pid=0;
+],[has_ucred=yes],[has_ucred=no])
+AC_MSG_RESULT($has_ucred)
+if test ${has_ucred} = no ; then CFLAGS="$CFLAGS -DNEED_UCRED"; fi
+
 dnl ----------------------------------------------------------------
 dnl Checks for libraries
 
--- openldap-2.3.39/libraries/liblutil/getpeereid.c	2007-06-10 18:39:53.000000000 +0000
+++ openldap-2.3.39/libraries/liblutil/getpeereid.c.new	2008-05-23 20:43:07.000000000 +0000
@@ -49,6 +49,15 @@
 {
 #ifdef LDAP_PF_LOCAL
 #if defined( SO_PEERCRED )
+#ifdef NEED_UCRED /* from autoconf */
+struct ucred
+   {
+      pid_t pid; /* PID of sending process. */
+      uid_t uid; /* UID of sending process. */
+      gid_t gid; /* GID of sending process. */
+   };
+#endif /* NEED_UCRED */
+
 	struct ucred peercred;
 	ber_socklen_t peercredlen = sizeof peercred;
 
