diff -urp libtool-2.4.6.orig/build-aux/ltmain.in libtool-2.4.6/build-aux/ltmain.in
--- libtool-2.4.6.orig/build-aux/ltmain.in	2015-02-06 10:57:56.000000000 -0200
+++ libtool-2.4.6/build-aux/ltmain.in	2016-12-23 11:57:43.709751017 -0200
@@ -2373,7 +2373,7 @@ func_mode_install ()
 	  # are installed into $libdir/../bin (currently, that works fine)
 	  # but it's something to keep an eye on.
 	  test "$inst_prefix_dir" = "$destdir" && \
-	    func_fatal_error "error: cannot install '$file' to a directory not ending in $libdir"
+	    func_echo "GoboLinux: installing '$file' to $destdir"
 
 	  if test -n "$inst_prefix_dir"; then
 	    # Stick the inst_prefix_dir data into the link command.
diff -urp libtool-2.4.6.orig/build-aux/ltmain.sh libtool-2.4.6/build-aux/ltmain.sh
--- libtool-2.4.6.orig/build-aux/ltmain.sh	2015-02-15 14:15:12.000000000 -0200
+++ libtool-2.4.6/build-aux/ltmain.sh	2016-12-23 11:58:08.644751484 -0200
@@ -4285,7 +4285,7 @@ func_mode_install ()
 	  # are installed into $libdir/../bin (currently, that works fine)
 	  # but it's something to keep an eye on.
 	  test "$inst_prefix_dir" = "$destdir" && \
-	    func_fatal_error "error: cannot install '$file' to a directory not ending in $libdir"
+	    func_echo "GoboLinux: installing '$file' to $destdir"
 
 	  if test -n "$inst_prefix_dir"; then
 	    # Stick the inst_prefix_dir data into the link command.
