--- GoboHide-0.09/Makefile.orig	2004-12-20 00:24:31.000000000 -0200
+++ GoboHide-0.09/Makefile	2004-12-20 00:27:36.000000000 -0200
@@ -1,9 +1,11 @@
 #*********************************************************************
 # Only change stuff from here if you know what you're doing
 CC = gcc
+INSTALL = install
 
 CFLAGS = -I./include
 LDFLAGS =
+PREFIX = /usr
 
 all: gobohide
 
@@ -11,9 +13,15 @@
 
 clean:
 	-rm -f *.o *~ gobohide
+
+install:
+	$(INSTALL) gobohide $(PREFIX)/sbin
+	$(INSTALL) gobohide.8 $(PREFIX)/man/man8
+	$(INSTALL) hier.7 $(PREFIX)/man/man7
 	
 gobohide: gobohide.o
 	$(CC) -o gobohide gobohide.o $(LDFLAGS)
 
 gobohide.o: gobohide.c
 	$(CC) -c gobohide.c $(CFLAGS)
+
