# fix: compile in kernel >= 2.6.26
diff -ur zaptel-1.2.27-patch/xpp/xbus-sysfs.c zaptel-1.2.27/xpp/xbus-sysfs.c
--- zaptel-1.2.27-patch/xpp/xbus-sysfs.c	2010-01-21 13:29:11.000000000 -0200
+++ zaptel-1.2.27/xpp/xbus-sysfs.c	2010-01-21 13:26:29.000000000 -0200
@@ -72,7 +72,7 @@
 /*--------- Sysfs Bus handling ----*/
 static int xpp_bus_match(struct device *dev, struct device_driver *driver)
 {
-	DBG(GENERAL, "dev->bus_id = %s, driver->name = %s\n", dev->bus_id, driver->name);
+	DBG(GENERAL, "dev->bus_id = %s, driver->name = %s\n", dev_name(dev), driver->name);
 	return 1;
 }
 
@@ -119,7 +119,7 @@
 	if(!dev)
 		return -ENODEV;
 	xbus = dev_to_xbus(dev);
-	DBG(GENERAL, "bus_id=%s xbus=%s\n", dev->bus_id, xbus->busname);
+	DBG(GENERAL, "bus_id=%s xbus=%s\n", dev_name(dev), xbus->busname);
 	XBUS_VAR_BLOCK;
 	envp[i] = NULL;
 	return 0;
@@ -140,7 +140,7 @@
 	if(!dev)
 		return -ENODEV;
 	xbus = dev_to_xbus(dev);
-	DBG(GENERAL, "bus_id=%s xbus=%s\n", dev->bus_id, xbus->busname);
+	DBG(GENERAL, "bus_id=%s xbus=%s\n", dev_name(dev), xbus->busname);
 	XBUS_VAR_BLOCK;
 	return 0;
 }
@@ -173,7 +173,7 @@
 };
 
 static struct device xpp_bus = {
-	.bus_id		= "xppbus",
+//	.bus_id		= "xppbus",
 	.release	= xpp_bus_release
 };
 
@@ -381,7 +381,7 @@
 	device_initialize(astribank);
 	astribank->bus = &xpp_bus_type;
 	astribank->parent = &xpp_bus;
-	snprintf(astribank->bus_id, BUS_ID_SIZE, "xbus-%02d", xbus->num);
+	dev_set_name(astribank, "xbus-%02d", xbus->num);
 	astribank->driver_data = NULL;	/* override below */
 	astribank->release = xpp_dev_release;
 	ret = device_register(astribank);
diff -ur zaptel-1.2.27-orig/wctdm24xxp/base.c zaptel-1.2.27/wctdm24xxp/base.c
--- zaptel-1.2.27-orig/wctdm24xxp/base.c	2008-04-04 14:46:10.000000000 -0300
+++ zaptel-1.2.27/wctdm24xxp/base.c	2010-01-21 12:32:16.000000000 -0200
@@ -35,7 +35,11 @@
 #include <linux/interrupt.h>
 #include <linux/workqueue.h>
 #include <linux/delay.h>
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,26)
+#include <linux/semaphore.h>
+#else
 #include <asm/semaphore.h>
+#endif
 
 #ifdef LINUX26
 #include <linux/moduleparam.h>
diff -ur zaptel-1.2.27-orig/wctdm24xxp/GpakApi.c zaptel-1.2.27/wctdm24xxp/GpakApi.c
--- zaptel-1.2.27-orig/wctdm24xxp/GpakApi.c	2008-04-04 12:59:39.000000000 -0300
+++ zaptel-1.2.27/wctdm24xxp/GpakApi.c	2010-01-21 12:32:16.000000000 -0200
@@ -34,7 +34,11 @@
  */
 
 #include <linux/version.h>
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,26)
+#include <linux/semaphore.h>
+#else
 #include <asm/semaphore.h>
+#endif
 
 #include "zaptel.h"
 
diff -ur zaptel-1.2.27-orig/wctdm24xxp/GpakCust.c zaptel-1.2.27/wctdm24xxp/GpakCust.c
--- zaptel-1.2.27-orig/wctdm24xxp/GpakCust.c	2008-01-03 15:13:22.000000000 -0200
+++ zaptel-1.2.27/wctdm24xxp/GpakCust.c	2010-01-21 12:32:16.000000000 -0200
@@ -37,7 +37,11 @@
 
 #include <linux/version.h>
 #include <linux/delay.h>
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,26)
+#include <linux/semaphore.h>
+#else
 #include <asm/semaphore.h>
+#endif
 
 #include "zaptel.h"
 #include "wctdm24xxp.h"
diff -ur zaptel-1.2.27-orig/wctdm24xxp/wctdm24xxp.h zaptel-1.2.27/wctdm24xxp/wctdm24xxp.h
--- zaptel-1.2.27-orig/wctdm24xxp/wctdm24xxp.h	2008-07-07 14:30:37.000000000 -0300
+++ zaptel-1.2.27/wctdm24xxp/wctdm24xxp.h	2010-01-21 12:32:16.000000000 -0200
@@ -29,7 +29,12 @@
 
 #include "../zaptel.h"
 #include "../voicebus.h"
+#include <linux/version.h>
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,26)
+#include <linux/semaphore.h>
+#else
 #include <asm/semaphore.h>
+#endif
 
 #define NUM_FXO_REGS 60
 
diff -ur zaptel-1.2.27-orig/wcte12xp/vpmadt032.c zaptel-1.2.27/wcte12xp/vpmadt032.c
--- zaptel-1.2.27-orig/wcte12xp/vpmadt032.c	2008-04-04 12:59:39.000000000 -0300
+++ zaptel-1.2.27/wcte12xp/vpmadt032.c	2010-01-21 12:32:16.000000000 -0200
@@ -29,7 +29,12 @@
  */
 
 #include <linux/delay.h>
+#include <linux/version.h>
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,26)
+#include <linux/semaphore.h>
+#else
 #include <asm/semaphore.h>
+#endif
 #include <linux/pci.h> 
 #include <linux/firmware.h> 
 #include <linux/list.h> 
diff -ur zaptel-1.2.27-orig/xpp/xpd.h zaptel-1.2.27/xpp/xpd.h
--- zaptel-1.2.27-orig/xpp/xpd.h	2008-05-22 11:56:10.000000000 -0300
+++ zaptel-1.2.27/xpp/xpd.h	2010-01-21 12:32:16.000000000 -0200
@@ -29,8 +29,13 @@
 #ifdef	__KERNEL__
 #include <linux/kernel.h>
 #include <linux/device.h>
+#include <linux/version.h>
 #include <asm/atomic.h>
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,26)
+#include <linux/semaphore.h>
+#else
 #include <asm/semaphore.h>
+#endif
 #include <linux/moduleparam.h>
 #ifdef	XPP_DEBUGFS
 #ifndef	CONFIG_DEBUG_FS
diff -ur zaptel-1.2.27-orig/wctc4xxp/base.c zaptel-1.2.27/wctc4xxp/base.c
--- zaptel-1.2.27-orig/wctc4xxp/base.c	2008-01-25 21:59:37.000000000 -0200
+++ zaptel-1.2.27/wctc4xxp/base.c	2010-01-21 12:55:03.000000000 -0200
@@ -38,7 +38,12 @@
 #include <linux/mman.h>
 #include <linux/delay.h>
 #include <asm/io.h>
+#include <linux/version.h>
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,26)
+#include <linux/semaphore.h>
+#else
 #include <asm/semaphore.h>
+#endif
 #include <linux/jiffies.h>
 #include <linux/workqueue.h>
 #include <linux/moduleparam.h>
