From 4d7cd8cec7ba56230b298986d7a91a7587984bc4 Mon Sep 17 00:00:00 2001
From: Remi Cardona <remi@gentoo.org>
Date: Sun, 19 Oct 2008 23:45:14 +0200
Subject: [PATCH] never enable DRI_MM on this branch

---
 configure.ac |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/configure.ac b/configure.ac
index 809e84d..04e31f3 100644
--- a/configure.ac
+++ b/configure.ac
@@ -202,7 +202,7 @@ if test "$DRI" = yes; then
         PKG_CHECK_MODULES(DRI, [libdrm xf86driproto glproto])
         AC_DEFINE(XF86DRI,1,[Enable DRI driver support])
         AC_DEFINE(XF86DRI_DEVEL,1,[Enable developmental DRI driver support])
-	PKG_CHECK_MODULES(DRI_MM, [libdrm >= 2.4.0],[DRI_MM=yes], [DRI_MM=no])
+	DRI_MM="no"
 	if test "x$DRI_MM" = xyes; then
 		AC_DEFINE(XF86DRI_MM,1,[Extended DRI memory management])
 	fi
-- 
1.6.0.2

