Don't run depmod after kernel module is installed

Signed-off-by: Jonas Karlsson <jonka750@student.liu.se>

--- rt73-cvs-2008011908/Module/Makefile	2008-01-17 18:43:35.000000000 +0100
+++ rt73-cvs-2008011908/Module/Makefile.new	2008-01-19 16:25:55.000000000 +0100
@@ -152,13 +152,8 @@
 		exit 1; \
 	fi
 	@echo "*** Install module in $(MODULE_ROOT) ..."
-ifeq ($(PATCHLEVEL),4)
 	install -m 755 -o 0 -g 0 -d $(MODULE_ROOT)
 	install -m 644 -o 0 -g 0 $(MODULE_OBJECT) $(MODULE_ROOT)
-else
-	@$(MAKE) $(KBUILD_PARAMS) modules_install
-endif
-	/sbin/depmod -a
 
 install: modules_install
 	@echo "*** Update $(MODULE_CONF) alias for $(IF_NAME)"; \
