diff -Puar pmount-0.9.3-orig/Makefile pmount-0.9.3/Makefile
--- pmount-0.9.3-orig/Makefile	2005-06-27 06:16:23.000000000 -0700
+++ pmount-0.9.3/Makefile	2005-10-18 12:54:54.000000000 -0700
@@ -25,13 +25,13 @@
 	$(CC) $(LDFLAGS) $(HAL_LDFLAGS) $^ $(LIBS) -o $@
 
 install: all install-mo
-	install -o root -g root -m 4755 -D ./pmount $(DESTDIR)/$(PREFIX)/bin/pmount
-	install -o root -g root -m 4755 -D ./pumount $(DESTDIR)/$(PREFIX)/bin/pumount
-	install -o root -g root -m 755 -D ./pmount-hal $(DESTDIR)/$(PREFIX)/bin/pmount-hal
-	install -o root -g root -m 644 -D ./pmount.1 $(DESTDIR)/$(PREFIX)/share/man/man1/pmount.1
-	install -o root -g root -m 644 -D ./pumount.1 $(DESTDIR)/$(PREFIX)/share/man/man1/pumount.1
-	install -o root -g root -m 644 -D ./pmount-hal.1 $(DESTDIR)/$(PREFIX)/share/man/man1/pmount-hal.1
-	install -o root -g root -m 644 -D ./pmount.allow $(DESTDIR)/$(SYSCONF)/pmount.allow
+	install -o 0 -g 0 -m 4755 -D ./pmount $(DESTDIR)/$(PREFIX)/bin/pmount
+	install -o 0 -g 0 -m 4755 -D ./pumount $(DESTDIR)/$(PREFIX)/bin/pumount
+	install -o 0 -g 0 -m 755 -D ./pmount-hal $(DESTDIR)/$(PREFIX)/bin/pmount-hal
+	install -o 0 -g 0 -m 644 -D ./pmount.1 $(DESTDIR)/$(PREFIX)/share/man/man1/pmount.1
+	install -o 0 -g 0 -m 644 -D ./pumount.1 $(DESTDIR)/$(PREFIX)/share/man/man1/pumount.1
+	install -o 0 -g 0 -m 644 -D ./pmount-hal.1 $(DESTDIR)/$(PREFIX)/share/man/man1/pmount-hal.1
+	install -o 0 -g 0 -m 644 -D ./pmount.allow $(DESTDIR)/$(SYSCONF)/pmount.allow
 
 uninstall: uninstall-mo
 	rm -f $(DESTDIR)/$(PREFIX)/bin/pmount $(DESTDIR)/$(PREFIX)/bin/pumount $(DESTDIR)/$(PREFIX)/bin/pmount-hal
