--- nss-3.11.4/mozilla/security/nss/cmd/shlibsign/sign.sh.orig	2007-01-15 10:31:44.000000000 -0200
+++ nss-3.11.4/mozilla/security/nss/cmd/shlibsign/sign.sh	2007-01-15 10:45:06.000000000 -0200
@@ -18,8 +18,8 @@ WIN*)
         PATH=${ARG1}/lib:${ARG1}/bin:${ARG4}:${PATH}
     fi
     export PATH
-    echo ${2}/shlibsign -v -i ${5}
-    ${2}/shlibsign -v -i ${5}
+    echo ${SHLIBSIGN} -v -i ${5}
+    ${SHLIBSIGN} -v -i ${5}
     ;;
 OpenVMS)
     temp="tmp$$.tmp"
@@ -31,8 +31,8 @@ OpenVMS)
     echo '$ define/job getipnodebyname xxx' >> $temp2
     echo '$ define/job vms_null_dl_name sys$share:decc$shr' >> $temp2
     dcl @$temp2
-    echo ${2}/shlibsign -v -i ${5}
-    ${2}/shlibsign -v -i ${5}
+    echo ${SHLIBSIGN} -v -i ${5}
+    ${SHLIBSIGN} -v -i ${5}
     sed -e "s/\([^\.]*\)\.so/\$ deass\/job \1/" $temp > $temp2
     echo '$ deass/job getipnodebyname' >> $temp2
     echo '$ deass/job vms_null_dl_name' >> $temp2
@@ -44,15 +44,9 @@ OpenVMS)
     export LIBPATH
     SHLIB_PATH=${1}/lib:${4}:$SHLIB_PATH
     export SHLIB_PATH
-    LD_LIBRARY_PATH=${1}/lib:${4}:$LD_LIBRARY_PATH
-    export LD_LIBRARY_PATH
-    DYLD_LIBRARY_PATH=${1}/lib:${4}:$DYLD_LIBRARY_PATH
-    export DYLD_LIBRARY_PATH
-    LIBRARY_PATH=${1}/lib:${4}:$LIBRARY_PATH
-    export LIBRARY_PATH
     ADDON_PATH=${1}/lib:${4}:$ADDON_PATH
     export ADDON_PATH
-    echo ${2}/shlibsign -v -i ${5}
-    ${2}/shlibsign -v -i ${5}
+    echo ${SHLIBSIGN} -v -i ${5}
+    ${SHLIBSIGN} -v -i ${5}
     ;;
 esac
