diff -Nur bc-1.06-old/bc/scan.c bc-1.06/bc/scan.c
--- bc-1.06-old/bc/scan.c	2000-10-03 20:35:13.000000000 +0200
+++ bc-1.06/bc/scan.c	2005-08-12 22:24:42.000000000 +0200
@@ -731,7 +731,6 @@
 
 /* Definitions for readline access. */
 extern FILE *rl_instream;
-_PROTOTYPE(char *readline, (char *));
 
 /* rl_input puts upto MAX characters into BUF with the number put in
    BUF placed in *RESULT.  If the yy input file is the same as
diff -Nur bc-1.06-old/bc/scan.l bc-1.06/bc/scan.l
--- bc-1.06-old/bc/scan.l	2000-09-13 20:25:47.000000000 +0200
+++ bc-1.06/bc/scan.l	2005-08-12 22:24:56.000000000 +0200
@@ -143,7 +143,6 @@
 
 /* Definitions for readline access. */
 extern FILE *rl_instream;
-_PROTOTYPE(char *readline, (char *));
 
 /* rl_input puts upto MAX characters into BUF with the number put in
    BUF placed in *RESULT.  If the yy input file is the same as
