--- minicom-2.1/src/window.c.orig	2006-08-16 14:32:02.000000000 -0300
+++ minicom-2.1/src/window.c	2006-08-16 14:33:59.000000000 -0300
@@ -62,7 +62,7 @@ static char *CE, *Al, *Dl, *AL, *DL;
 static char *CS, *SF, *SR, *VB, *BL;
 static char *VE, *VI, *KS, *KE;
 static char *CD, *CL, *IC, *DC;
-static char *BC, *CR, *NL;
+static char *this_BC, *CR, *NL;
 #if ST_LINE
 static char *TS, *FS, *DS;
 #endif
@@ -322,8 +322,8 @@ int x, y;
   else if (NL != CNULL && x == 0 && x == curx && y == cury + 1)
 	outstr(NL);
 #endif
-  else if (BC != CNULL && y == cury && x == curx - 1)
-  	outstr(BC);
+  else if (this_BC != CNULL && y == cury && x == curx - 1)
+  	outstr(this_BC);
   else	
   	outstr(tgoto(CM, x, y));
   curx = x;
@@ -1918,7 +1918,7 @@ int attr;
   CL = tgetstr("cl", &_tptr);
   IC = tgetstr("ic", &_tptr);
   DC = tgetstr("dc", &_tptr);
-  BC = tgetstr("bc", &_tptr);
+  this_BC = tgetstr("bc", &_tptr);
   CR = tgetstr("cr", &_tptr);
   NL = tgetstr("nl", &_tptr);
   AC = tgetstr("ac", &_tptr);
@@ -1967,10 +1967,10 @@ int attr;
   _has_am = tgetflag("am");
   _mv_standout = tgetflag("ms");
   if (tgetflag("bs")) {
-    if (BC == CNULL) BC = "\b";
+    if (this_BC == CNULL) this_BC = "\b";
   } 
   else
-    BC = CNULL;	
+    this_BC = CNULL;	
 
   /* Special IBM box-drawing characters */
   D_UL  = 201;
