diff -Naur kbd-1.08/src/Makefile.in kbd-1.08-patched/src/Makefile.in
--- kbd-1.08/src/Makefile.in	Sun Oct 13 03:34:10 2002
+++ kbd-1.08-patched/src/Makefile.in	Tue Oct 15 17:24:22 2002
@@ -37,8 +37,9 @@
 OLDPROGS= mapscrn loadunimap
 
 # Not installed by default
-MISC    = screendump setlogcons setvesablank spawn_console spawn_login \
-	  getunimap clrunimap outpsfheader setpalette
+#MISC    = screendump setlogcons setvesablank spawn_console spawn_login \
+#	  getunimap clrunimap outpsfheader setpalette
+MISC     = setlogcons setvesablank getunimap
 
 # Installed by default
 SHCMDS  = unicode_start unicode_stop
@@ -64,7 +65,7 @@
 install:	all
 	install -d -m 755 $(BINDIR) $(LOADKEYS_BINDIR)
 	install -s -m 0755 -o root $(PROGS) $(OLDPROGS) $(BINDIR)
-#	install -s -m 0755 -o root $(MISC) $(BINDIR)
+	install -s -m 0755 -o root $(MISC) $(BINDIR)
 	install -c -m 0755 -o root $(SHCMDS) $(BINDIR)
 	for i in psfaddtable psfgettable psfstriptable; do \
 		rm -f $(BINDIR)/$$i; ln -s psfxtable $(BINDIR)/$$i; \
